02.10-bismark-deduplication
================
Sam White
2024-11-09

- <a href="#1-intro" id="toc-1-intro">1 Intro</a>
  - <a href="#11-inputs" id="toc-11-inputs">1.1 Inputs</a>
  - <a href="#12-outputs" id="toc-12-outputs">1.2 Outputs</a>
- <a href="#2-create-a-bash-variables-file"
  id="toc-2-create-a-bash-variables-file">2 Create a Bash variables
  file</a>
- <a href="#3-deduplication" id="toc-3-deduplication">3 Deduplication</a>
- <a href="#4-sorting" id="toc-4-sorting">4 Sorting</a>
  - <a href="#41-check-file-counts" id="toc-41-check-file-counts">4.1 Check
    file counts</a>
- <a href="#5-indexing-for-downstream-applications"
  id="toc-5-indexing-for-downstream-applications">5 Indexing for
  Downstream Applications</a>
- <a href="#6-checksums" id="toc-6-checksums">6 Checksums</a>
- <a href="#7-multiqc" id="toc-7-multiqc">7 MultiQC</a>
- <a href="#8-list-output-files" id="toc-8-list-output-files">8 List
  output files</a>

# 1 Intro

## 1.1 Inputs

This notebook is a continuation of using
[Bismark](https://felixkrueger.github.io/Bismark/) (Krueger and Andrews
2011) to perform methylation analysis. Read alignments were run in
[02.00-bismark-bowtie2-alignment.Rmd](./02.00-bismark-bowtie2-alignment.Rmd).
This step will perform
[deduplication](https://felixkrueger.github.io/Bismark/bismark/deduplication/)
(Bismark manual) of the BAM files generated by alignment. Results will
be summarized by [MultiQC](https://github.com/MultiQC/MultiQC) (Ewels et
al. 2016).

Expects input BAM filenames formatted like (e.g.):

- `CF03-CM03-Zygote_R1_001.fastp-trim_bismark_bt2_pe.bam`

## 1.2 Outputs

- `*_dedup.sorted.bam`: Deduplicated, sorted BAMs.

- `*_dedup.sorted.bam.bai`: Deduplicated, sorted BAM index file. Useful
  for IGV.

- `*.deduplicated.bam`: Deduplicated BAMs. These are *not* sorted.

- `*.deduplication_report.txt`: Individual summary reports for each
  deduplication process.

- `multiqc_report.html`: A summary report of the alignment results
  generated by [MultiQC](https://github.com/MultiQC/MultiQC), in HTML
  format.

Due to the large file sizes of BAMS, these cannot be hosted in the
[ceasmallr GitHub repo](https://github.com/sr320/ceasmallr). As such
these files are available for download here:

- <https://gannet.fish.washington.edu/gitrepos/ceasmallr/output/02.10-bismark-deduplication/>

# 2 Create a Bash variables file

This allows usage of Bash variables across R Markdown chunks.

``` bash
{
echo "#### Assign Variables ####"
echo ""

echo "# Data directories"
echo 'export repo_dir=/home/shared/8TB_HDD_01/sam/gitrepos/ceasmallr'
echo 'export output_dir_top=${repo_dir}/output/02.10-bismark-deduplication'
echo 'export alignment_dir="${repo_dir}/output/02.01-bismark-bowtie2-alignment-SLURM-array"'
echo ""


echo "# Paths to programs"
echo 'export programs_dir="/home/shared"'
echo 'export bismark_dir="${programs_dir}/Bismark-0.24.0"'
echo 'export bowtie2_dir="${programs_dir}/bowtie2-2.4.4-linux-x86_64"'
echo 'export multiqc="/home/sam/programs/mambaforge/bin/multiqc"'
echo 'export samtools_dir="${programs_dir}/samtools-1.12"'
echo ""

echo "# Set number of CPUs to use"
echo 'export threads=40'
echo ""

echo "# Filename patterns"
echo "export deduped_filename_pattern='_R1_001.fastp-trim*bismark_bt2_pe.deduplicated.bam'"


echo "# Print formatting"
echo 'export line="--------------------------------------------------------"'
echo ""
} > .bashvars

cat .bashvars
```

    #### Assign Variables ####

    # Data directories
    export repo_dir=/home/shared/8TB_HDD_01/sam/gitrepos/ceasmallr
    export output_dir_top=${repo_dir}/output/02.10-bismark-deduplication
    export alignment_dir="${repo_dir}/output/02.01-bismark-bowtie2-alignment-SLURM-array"

    # Paths to programs
    export programs_dir="/home/shared"
    export bismark_dir="${programs_dir}/Bismark-0.24.0"
    export bowtie2_dir="${programs_dir}/bowtie2-2.4.4-linux-x86_64"
    export multiqc="/home/sam/programs/mambaforge/bin/multiqc"
    export samtools_dir="${programs_dir}/samtools-1.12"

    # Set number of CPUs to use
    export threads=40

    # Filename patterns
    export deduped_filename_pattern='_R1_001.fastp-trim*bismark_bt2_pe.deduplicated.bam'
    # Print formatting
    export line="--------------------------------------------------------"

# 3 Deduplication

``` bash
# Load bash variables into memory
source .bashvars

# Make output directory, if it doesn't exist
mkdir --parents "${output_dir_top}"

cd "${alignment_dir}"

for bam in *_bismark_bt2_pe.bam
do
  # Set stderr filename
  stderr_name=$(echo "${bam}" | awk -F"." '{print $1"."$2".deduplication.stderr"}')
  
  # Run deduplication
  ${bismark_dir}/deduplicate_bismark \
  --samtools_path ${samtools_dir} \
  --paired \
  --bam \
  --output_dir ${output_dir_top} \
  ${bam} \
  > ${output_dir_top}/${stderr_name} 2>&1
done
```

# 4 Sorting

``` bash
# Load bash variables into memory
source .bashvars

cd "${output_dir_top}"


for bam in *deduplicated.bam
do
  sample=$(echo ${bam} | awk -F"_" '{print $1}')
  ${samtools_dir}/samtools sort \
  -@ "${threads}" \
  -o ${sample}_dedup.sorted.bam \
  ${bam}
done
```

## 4.1 Check file counts

``` bash
# Load bash variables into memory
source .bashvars

cd "${output_dir_top}"
ls -1 *deduplicated.bam | wc -l
ls -1 *dedup.sorted.bam | wc -l
```

# 5 Indexing for Downstream Applications

``` bash
# Load bash variables into memory
source .bashvars

cd "${output_dir_top}"
find *dedup.sorted.bam \
| xargs basename -s _dedup.sorted.bam \
| xargs -I{} ${samtools_dir}/samtools \
index {}_dedup.sorted.bam
```

# 6 Checksums

``` bash
# Load bash variables into memory
source .bashvars

cd "${output_dir_top}"

for file in *; do
  if [ "${file}" != "checksums.md5" ]; then
    md5sum "${file}" >> checksums.md5
  fi
done
```

# 7 MultiQC

``` bash
# Load bash variables into memory
source .bashvars

cd "${output_dir_top}"

${multiqc} .
```

# 8 List output files

``` bash
# Load bash variables into memory
source .bashvars

cd "${output_dir_top}"

ls -lh
```

    total 238G
    -rw-r--r-- 1 sam sam  520M Nov 26 07:36 CF01-CM01-Zygote_dedup.sorted.bam
    -rw-rw-r-- 1 sam sam  604K Nov 26 09:55 CF01-CM01-Zygote_dedup.sorted.bam.bai
    -rw-rw-r-- 1 sam sam  3.3K Nov 26 09:55 CF01-CM01-Zygote_R1_001.fastp-trim.deduplication.stderr
    -rw-r--r-- 1 sam sam  578M Nov 25 12:07 CF01-CM01-Zygote_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.bam
    -rw-rw-r-- 1 sam sam   315 Nov 26 09:55 CF01-CM01-Zygote_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplication_report.txt
    -rw-r--r-- 1 sam sam  4.9G Nov 26 07:39 CF01-CM02-Larvae_dedup.sorted.bam
    -rw-rw-r-- 1 sam sam  1.5M Nov 26 09:55 CF01-CM02-Larvae_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam  6.7G Nov 25 12:25 CF01-CM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bam
    -rw-rw-r-- 1 sam sam   310 Nov 26 09:55 CF01-CM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplication_report.txt
    -rw-rw-r-- 1 sam sam  3.2K Nov 26 09:55 CF01-CM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam  2.3G Nov 26 07:41 CF02-CM02-Zygote_dedup.sorted.bam
    -rw-rw-r-- 1 sam sam  1.1M Nov 26 09:55 CF02-CM02-Zygote_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam  2.9G Nov 25 12:32 CF02-CM02-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bam
    -rw-rw-r-- 1 sam sam   310 Nov 26 09:55 CF02-CM02-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplication_report.txt
    -rw-rw-r-- 1 sam sam  3.2K Nov 26 09:55 CF02-CM02-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam  3.0G Nov 26 07:42 CF03-CM03-Zygote_dedup.sorted.bam
    -rw-rw-r-- 1 sam sam  1.2M Nov 26 09:55 CF03-CM03-Zygote_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam  3.8G Nov 25 12:42 CF03-CM03-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bam
    -rw-rw-r-- 1 sam sam   309 Nov 26 09:55 CF03-CM03-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplication_report.txt
    -rw-rw-r-- 1 sam sam  3.2K Nov 26 09:55 CF03-CM03-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam  3.6G Nov 26 07:44 CF03-CM04-Larvae_dedup.sorted.bam
    -rw-rw-r-- 1 sam sam  1.3M Nov 26 09:55 CF03-CM04-Larvae_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam  5.1G Nov 25 12:57 CF03-CM04-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bam
    -rw-rw-r-- 1 sam sam   311 Nov 26 09:55 CF03-CM04-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplication_report.txt
    -rw-rw-r-- 1 sam sam  3.2K Nov 26 09:55 CF03-CM04-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam  4.5G Nov 26 07:46 CF03-CM05-Larvae_dedup.sorted.bam
    -rw-rw-r-- 1 sam sam  1.5M Nov 26 09:55 CF03-CM05-Larvae_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam  6.0G Nov 25 13:13 CF03-CM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bam
    -rw-rw-r-- 1 sam sam   309 Nov 26 09:55 CF03-CM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplication_report.txt
    -rw-rw-r-- 1 sam sam  3.2K Nov 26 09:55 CF03-CM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam  3.3G Nov 26 07:48 CF04-CM04-Zygote_dedup.sorted.bam
    -rw-rw-r-- 1 sam sam  1.3M Nov 26 09:55 CF04-CM04-Zygote_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam  4.3G Nov 25 13:24 CF04-CM04-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bam
    -rw-rw-r-- 1 sam sam   310 Nov 26 09:55 CF04-CM04-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplication_report.txt
    -rw-rw-r-- 1 sam sam  3.2K Nov 26 09:55 CF04-CM04-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam  5.9G Nov 26 07:53 CF05-CM02-Larvae_dedup.sorted.bam
    -rw-rw-r-- 1 sam sam  1.6M Nov 26 09:55 CF05-CM02-Larvae_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam  8.4G Nov 25 13:47 CF05-CM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bam
    -rw-rw-r-- 1 sam sam   311 Nov 26 09:55 CF05-CM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplication_report.txt
    -rw-rw-r-- 1 sam sam  3.2K Nov 26 09:55 CF05-CM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam  3.9G Nov 26 07:55 CF05-CM05-Zygote_dedup.sorted.bam
    -rw-rw-r-- 1 sam sam  1.4M Nov 26 09:55 CF05-CM05-Zygote_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam  5.1G Nov 25 14:01 CF05-CM05-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bam
    -rw-rw-r-- 1 sam sam   310 Nov 26 09:55 CF05-CM05-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplication_report.txt
    -rw-rw-r-- 1 sam sam  3.2K Nov 26 09:55 CF05-CM05-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam  3.7G Nov 26 07:57 CF06-CM01-Zygote_dedup.sorted.bam
    -rw-rw-r-- 1 sam sam  1.4M Nov 26 09:55 CF06-CM01-Zygote_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam  4.7G Nov 25 14:13 CF06-CM01-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bam
    -rw-rw-r-- 1 sam sam   309 Nov 26 09:55 CF06-CM01-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplication_report.txt
    -rw-rw-r-- 1 sam sam  3.2K Nov 26 09:55 CF06-CM01-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam  2.9G Nov 26 07:58 CF06-CM02-Larvae_dedup.sorted.bam
    -rw-rw-r-- 1 sam sam  1.2M Nov 26 09:55 CF06-CM02-Larvae_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam  3.6G Nov 25 14:23 CF06-CM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bam
    -rw-rw-r-- 1 sam sam   309 Nov 26 09:55 CF06-CM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplication_report.txt
    -rw-rw-r-- 1 sam sam  3.2K Nov 26 09:55 CF06-CM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam  2.3G Nov 26 07:59 CF07-CM02-Zygote_dedup.sorted.bam
    -rw-rw-r-- 1 sam sam  1.1M Nov 26 09:55 CF07-CM02-Zygote_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam  2.8G Nov 25 14:30 CF07-CM02-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bam
    -rw-rw-r-- 1 sam sam   309 Nov 26 09:55 CF07-CM02-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplication_report.txt
    -rw-rw-r-- 1 sam sam  3.2K Nov 26 09:55 CF07-CM02-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam  2.4G Nov 26 08:00 CF08-CM03-Zygote_dedup.sorted.bam
    -rw-rw-r-- 1 sam sam  1.1M Nov 26 09:55 CF08-CM03-Zygote_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam  3.0G Nov 25 14:38 CF08-CM03-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bam
    -rw-rw-r-- 1 sam sam   309 Nov 26 09:55 CF08-CM03-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplication_report.txt
    -rw-rw-r-- 1 sam sam  3.2K Nov 26 09:55 CF08-CM03-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam   62M Nov 26 08:00 CF08-CM04-Larvae_dedup.sorted.bam
    -rw-rw-r-- 1 sam sam  1.1M Nov 26 09:55 CF08-CM04-Larvae_dedup.sorted.bam.bai
    -rw-rw-r-- 1 sam sam  3.3K Nov 26 09:55 CF08-CM04-Larvae_R1_001.fastp-trim.deduplication.stderr
    -rw-r--r-- 1 sam sam   65M Nov 25 14:38 CF08-CM04-Larvae_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.bam
    -rw-rw-r-- 1 sam sam   312 Nov 26 09:55 CF08-CM04-Larvae_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplication_report.txt
    -rw-r--r-- 1 sam sam   54M Nov 26 08:00 CF08-CM05-Larvae_dedup.sorted.bam
    -rw-rw-r-- 1 sam sam  1.1M Nov 26 09:55 CF08-CM05-Larvae_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam   58M Nov 25 14:38 CF08-CM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bam
    -rw-rw-r-- 1 sam sam   305 Nov 26 09:55 CF08-CM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplication_report.txt
    -rw-rw-r-- 1 sam sam  3.2K Nov 26 09:55 CF08-CM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplication.stderr
    -rw-rw-r-- 1 sam sam   15K Nov 26 09:55 checksums.md5
    -rw-r--r-- 1 sam sam  3.1G Nov 26 08:02 EF01-EM01-Zygote_dedup.sorted.bam
    -rw-rw-r-- 1 sam sam  1.3M Nov 26 09:55 EF01-EM01-Zygote_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam  4.0G Nov 25 14:49 EF01-EM01-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bam
    -rw-rw-r-- 1 sam sam   309 Nov 26 09:55 EF01-EM01-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplication_report.txt
    -rw-rw-r-- 1 sam sam  3.2K Nov 26 09:55 EF01-EM01-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam  3.3G Nov 26 08:03 EF02-EM02-Zygote_dedup.sorted.bam
    -rw-rw-r-- 1 sam sam  1.3M Nov 26 09:55 EF02-EM02-Zygote_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam  4.3G Nov 25 15:01 EF02-EM02-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bam
    -rw-rw-r-- 1 sam sam   309 Nov 26 09:55 EF02-EM02-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplication_report.txt
    -rw-rw-r-- 1 sam sam  3.2K Nov 26 09:55 EF02-EM02-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam  2.2G Nov 26 08:05 EF03-EM03-Zygote_dedup.sorted.bam
    -rw-rw-r-- 1 sam sam 1011K Nov 26 09:55 EF03-EM03-Zygote_dedup.sorted.bam.bai
    -rw-rw-r-- 1 sam sam  3.3K Nov 26 09:55 EF03-EM03-Zygote_R1_001.fastp-trim.deduplication.stderr
    -rw-r--r-- 1 sam sam  2.7G Nov 25 15:08 EF03-EM03-Zygote_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.bam
    -rw-rw-r-- 1 sam sam   318 Nov 26 09:55 EF03-EM03-Zygote_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplication_report.txt
    -rw-r--r-- 1 sam sam  4.6G Nov 26 08:07 EF03-EM04-Larvae_dedup.sorted.bam
    -rw-rw-r-- 1 sam sam  1.5M Nov 26 09:55 EF03-EM04-Larvae_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam  5.9G Nov 25 15:23 EF03-EM04-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bam
    -rw-rw-r-- 1 sam sam   310 Nov 26 09:55 EF03-EM04-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplication_report.txt
    -rw-rw-r-- 1 sam sam  3.2K Nov 26 09:55 EF03-EM04-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam  4.6G Nov 26 08:09 EF03-EM05-Larvae_dedup.sorted.bam
    -rw-rw-r-- 1 sam sam  1.5M Nov 26 09:55 EF03-EM05-Larvae_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam  6.1G Nov 25 15:39 EF03-EM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bam
    -rw-rw-r-- 1 sam sam   309 Nov 26 09:55 EF03-EM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplication_report.txt
    -rw-rw-r-- 1 sam sam  3.2K Nov 26 09:55 EF03-EM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam  1.3G Nov 26 08:10 EF04-EM04-Zygote_dedup.sorted.bam
    -rw-rw-r-- 1 sam sam  635K Nov 26 09:55 EF04-EM04-Zygote_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam  1.5G Nov 25 15:43 EF04-EM04-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bam
    -rw-rw-r-- 1 sam sam   308 Nov 26 09:55 EF04-EM04-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplication_report.txt
    -rw-rw-r-- 1 sam sam  3.2K Nov 26 09:55 EF04-EM04-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam  5.2G Nov 26 08:14 EF04-EM05-Larvae_dedup.sorted.bam
    -rw-rw-r-- 1 sam sam  1.6M Nov 26 09:55 EF04-EM05-Larvae_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam  7.5G Nov 25 16:04 EF04-EM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bam
    -rw-rw-r-- 1 sam sam   311 Nov 26 09:55 EF04-EM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplication_report.txt
    -rw-rw-r-- 1 sam sam  3.2K Nov 26 09:55 EF04-EM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam  3.4G Nov 26 08:16 EF05-EM01-Larvae_dedup.sorted.bam
    -rw-rw-r-- 1 sam sam  1.3M Nov 26 09:55 EF05-EM01-Larvae_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam  4.6G Nov 25 16:16 EF05-EM01-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bam
    -rw-rw-r-- 1 sam sam   309 Nov 26 09:55 EF05-EM01-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplication_report.txt
    -rw-rw-r-- 1 sam sam  3.2K Nov 26 09:55 EF05-EM01-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam  784M Nov 26 08:17 EF05-EM05-Zygote_dedup.sorted.bam
    -rw-rw-r-- 1 sam sam  580K Nov 26 09:55 EF05-EM05-Zygote_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam  860M Nov 25 16:18 EF05-EM05-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bam
    -rw-rw-r-- 1 sam sam   307 Nov 26 09:55 EF05-EM05-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplication_report.txt
    -rw-rw-r-- 1 sam sam  3.2K Nov 26 09:55 EF05-EM05-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam  5.3G Nov 26 08:22 EF05-EM06-Larvae_dedup.sorted.bam
    -rw-rw-r-- 1 sam sam  1.6M Nov 26 09:55 EF05-EM06-Larvae_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam  7.5G Nov 25 16:39 EF05-EM06-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bam
    -rw-rw-r-- 1 sam sam   311 Nov 26 09:55 EF05-EM06-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplication_report.txt
    -rw-rw-r-- 1 sam sam  3.2K Nov 26 09:55 EF05-EM06-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam  5.3G Nov 26 08:27 EF06-EM01-Larvae_dedup.sorted.bam
    -rw-rw-r-- 1 sam sam  1.6M Nov 26 09:55 EF06-EM01-Larvae_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam  7.5G Nov 25 17:00 EF06-EM01-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bam
    -rw-rw-r-- 1 sam sam   310 Nov 26 09:55 EF06-EM01-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplication_report.txt
    -rw-rw-r-- 1 sam sam  3.2K Nov 26 09:55 EF06-EM01-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam  4.4G Nov 26 08:30 EF06-EM02-Larvae_dedup.sorted.bam
    -rw-rw-r-- 1 sam sam  1.4M Nov 26 09:55 EF06-EM02-Larvae_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam  5.9G Nov 25 17:16 EF06-EM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bam
    -rw-rw-r-- 1 sam sam   310 Nov 26 09:55 EF06-EM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplication_report.txt
    -rw-rw-r-- 1 sam sam  3.2K Nov 26 09:55 EF06-EM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam  1.3G Nov 26 08:31 EF06-EM06-Larvae_dedup.sorted.bam
    -rw-rw-r-- 1 sam sam  644K Nov 26 09:55 EF06-EM06-Larvae_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam  1.6G Nov 25 17:23 EF06-EM06-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bam
    -rw-rw-r-- 1 sam sam   309 Nov 26 09:55 EF06-EM06-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplication_report.txt
    -rw-rw-r-- 1 sam sam  3.2K Nov 26 09:55 EF06-EM06-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam  1.6G Nov 26 08:31 EF07-EM01-Zygote_dedup.sorted.bam
    -rw-rw-r-- 1 sam sam  824K Nov 26 09:55 EF07-EM01-Zygote_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam  1.9G Nov 25 17:29 EF07-EM01-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bam
    -rw-rw-r-- 1 sam sam   310 Nov 26 09:55 EF07-EM01-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplication_report.txt
    -rw-rw-r-- 1 sam sam  3.2K Nov 26 09:55 EF07-EM01-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam  4.1G Nov 26 08:34 EF07-EM03-Larvae_dedup.sorted.bam
    -rw-rw-r-- 1 sam sam  1.4M Nov 26 09:55 EF07-EM03-Larvae_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam  5.8G Nov 25 17:45 EF07-EM03-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bam
    -rw-rw-r-- 1 sam sam   310 Nov 26 09:55 EF07-EM03-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplication_report.txt
    -rw-rw-r-- 1 sam sam  3.2K Nov 26 09:55 EF07-EM03-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam  4.6G Nov 26 08:37 EF08-EM03-Larvae_dedup.sorted.bam
    -rw-rw-r-- 1 sam sam  1.5M Nov 26 09:55 EF08-EM03-Larvae_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam  6.7G Nov 25 18:03 EF08-EM03-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bam
    -rw-rw-r-- 1 sam sam   311 Nov 26 09:55 EF08-EM03-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplication_report.txt
    -rw-rw-r-- 1 sam sam  3.2K Nov 26 09:55 EF08-EM03-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam  4.2G Nov 26 08:40 EF08-EM04-Larvae_dedup.sorted.bam
    -rw-rw-r-- 1 sam sam  1.4M Nov 26 09:55 EF08-EM04-Larvae_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam  5.6G Nov 25 18:18 EF08-EM04-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bam
    -rw-rw-r-- 1 sam sam   310 Nov 26 09:55 EF08-EM04-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplication_report.txt
    -rw-rw-r-- 1 sam sam  3.2K Nov 26 09:55 EF08-EM04-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplication.stderr
    drwxrwxr-x 2 sam sam  4.0K Nov 26 09:55 multiqc_data
    -rw-rw-r-- 1 sam sam  1.1M Nov 26 09:55 multiqc_report.html

<div id="refs" class="references csl-bib-body hanging-indent">

<div id="ref-ewels2016" class="csl-entry">

Ewels, Philip, Måns Magnusson, Sverker Lundin, and Max Käller. 2016.
“MultiQC: Summarize Analysis Results for Multiple Tools and Samples in a
Single Report.” *Bioinformatics* 32 (19): 3047–48.
<https://doi.org/10.1093/bioinformatics/btw354>.

</div>

<div id="ref-krueger2011" class="csl-entry">

Krueger, Felix, and Simon R. Andrews. 2011. “Bismark: A Flexible Aligner
and Methylation Caller for Bisulfite-Seq Applications.” *Bioinformatics*
27 (11): 1571–72. <https://doi.org/10.1093/bioinformatics/btr167>.

</div>

</div>
