02.10-bismark-deduplication
================
Sam White
2024-11-09

- <a href="#1-intro" id="toc-1-intro">1 Intro</a>
  - <a href="#11-inputs" id="toc-11-inputs">1.1 Inputs</a>
  - <a href="#12-outputs" id="toc-12-outputs">1.2 Outputs</a>
- <a href="#2-create-a-bash-variables-file"
  id="toc-2-create-a-bash-variables-file">2 Create a Bash variables
  file</a>
- <a href="#3-deduplication" id="toc-3-deduplication">3 Deduplication</a>
- <a href="#4-sorting" id="toc-4-sorting">4 Sorting</a>
- <a href="#5-indexing-for-downstream-applications"
  id="toc-5-indexing-for-downstream-applications">5 Indexing for
  Downstream Applications</a>
- <a href="#6-multiqc" id="toc-6-multiqc">6 MultiQC</a>
- <a href="#7-checksums" id="toc-7-checksums">7 Checksums</a>
- <a href="#8-list-output-files" id="toc-8-list-output-files">8 List
  output files</a>

# 1 Intro

## 1.1 Inputs

This notebook is a continuation of using
[Bismark](https://felixkrueger.github.io/Bismark/) (Krueger and Andrews
2011) to perform methylation analysis. Read alignments were run in
[02.00-bismark-bowtie2-alignment.Rmd](./02.00-bismark-bowtie2-alignment.Rmd).
This step will perform
[deduplication](https://felixkrueger.github.io/Bismark/bismark/deduplication/)
(Bismark manual) of the BAM files generated by alignment. Results will
be summarized by [MultiQC](https://github.com/MultiQC/MultiQC) (Ewels et
al. 2016).

## 1.2 Outputs

- `*_dedup.sorted.bam`: Deduplicated, sorted BAMs.

- `*_dedup.sorted.bam.bai`: Deduplicated, sorted BAM index file. Useful
  for IGV.

- `*.deduplicated.bam`: Deduplicated BAMs. These are *not* sorted.

- `*.deduplication_report.txt`: Individual summary reports for each
  deduplication process.

- `multiqc_report.html`: A summary report of the alignment results
  generated by [MultiQC](https://github.com/MultiQC/MultiQC), in HTML
  format.

Due to the large file sizes of BAMS, these cannot be hosted in the
[ceasmallr GitHub repo](https://github.com/sr320/ceasmallr). As such
these files are available for download here:

- <https://gannet.fish.washington.edu/Atumefaciens/gitrepos/ceasmallr/output/02.10-bismark-deduplication/>

# 2 Create a Bash variables file

This allows usage of Bash variables across R Markdown chunks.

``` bash
{
echo "#### Assign Variables ####"
echo ""

echo "# Data directories"
echo 'export repo_dir=/home/shared/8TB_HDD_01/sam/gitrepos/ceasmallr'
echo 'export output_dir_top=${repo_dir}/output/02.10-bismark-deduplication'
echo 'export alignment_dir="${repo_dir}/output/02.00-bismark-bowtie2-alignment"'
echo ""


echo "# Paths to programs"
echo 'export programs_dir="/home/shared"'
echo 'export bismark_dir="${programs_dir}/Bismark-0.24.0"'
echo 'export bowtie2_dir="${programs_dir}/bowtie2-2.4.4-linux-x86_64"'
echo 'export multiqc="/home/sam/programs/mambaforge/bin/multiqc"'
echo 'export samtools_dir="${programs_dir}/samtools-1.12"'
echo ""

echo "# Set number of CPUs to use"
echo 'export threads=40'


echo "# Print formatting"
echo 'export line="--------------------------------------------------------"'
echo ""
} > .bashvars

cat .bashvars
```

    #### Assign Variables ####

    # Data directories
    export repo_dir=/home/shared/8TB_HDD_01/sam/gitrepos/ceasmallr
    export output_dir_top=${repo_dir}/output/02.10-bismark-deduplication
    export alignment_dir="${repo_dir}/output/02.00-bismark-bowtie2-alignment"

    # Paths to programs
    export programs_dir="/home/shared"
    export bismark_dir="${programs_dir}/Bismark-0.24.0"
    export bowtie2_dir="${programs_dir}/bowtie2-2.4.4-linux-x86_64"
    export multiqc="/home/sam/programs/mambaforge/bin/multiqc"
    export samtools_dir="${programs_dir}/samtools-1.12"

    # Set number of CPUs to use
    export threads=40
    # Print formatting
    export line="--------------------------------------------------------"

# 3 Deduplication

``` bash
# Load bash variables into memory
source .bashvars

# Make output directory, if it doesn't exist
mkdir --parents "${output_dir_top}"

cd "${alignment_dir}"

for bam in *_bismark_bt2_pe.bam
do
  # Set stderr filename
  stderr_name=$(echo "${bam}" | awk -F"." '{print $1"."$2"."$3".deduplication.stderr"}')
  
  # Run deduplication
  ${bismark_dir}/deduplicate_bismark \
  --samtools_path ${samtools_dir} \
  --paired \
  --bam \
  --output_dir ${output_dir_top} \
  ${bam} \
  > ${output_dir_top}/${stderr_name} 2>&1
done
```

# 4 Sorting

``` bash
# Load bash variables into memory
source .bashvars

cd "${output_dir_top}"


#Sorting for Downstream Applications
find *deduplicated.bam \
| xargs basename -s _R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.bam \
| xargs -I{} ${samtools_dir}/samtools \
sort {}_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.bam \
-@ "${threads}" \
-o {}_dedup.sorted.bam
```

# 5 Indexing for Downstream Applications

``` bash
# Load bash variables into memory
source .bashvars

cd "${output_dir_top}"
find *dedup.sorted.bam \
| xargs basename -s _dedup.sorted.bam \
| xargs -I{} ${samtools_dir}/samtools \
index {}_dedup.sorted.bam
```

# 6 MultiQC

``` bash
# Load bash variables into memory
source .bashvars

cd "${output_dir_top}"

${multiqc} .
```

# 7 Checksums

``` bash
# Load bash variables into memory
source .bashvars

cd "${output_dir_top}"

for file in *; do
  if [ "${file}" != "checksums.md5" ]; then
    md5sum "${file}" >> checksums.md5
  fi
done
```

# 8 List output files

``` bash
# Load bash variables into memory
source .bashvars

cd "${output_dir_top}"

ls -lh
```

    total 213G
    -rw-r--r-- 1 sam sam 455M Nov 14 07:15 CF01-CM01-Zygote_dedup.sorted.bam
    -rw-r--r-- 1 sam sam 603K Nov 14 08:30 CF01-CM01-Zygote_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam 498M Nov 13 13:56 CF01-CM01-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.bam
    -rw-r--r-- 1 sam sam  315 Nov 13 13:56 CF01-CM01-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication_report.txt
    -rw-r--r-- 1 sam sam 3.4K Nov 13 13:56 CF01-CM01-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam 4.4G Nov 14 07:19 CF01-CM02-Larvae_dedup.sorted.bam
    -rw-r--r-- 1 sam sam 1.4M Nov 14 08:31 CF01-CM02-Larvae_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam 5.9G Nov 13 14:12 CF01-CM02-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.bam
    -rw-r--r-- 1 sam sam  319 Nov 13 14:12 CF01-CM02-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication_report.txt
    -rw-r--r-- 1 sam sam 3.4K Nov 13 14:12 CF01-CM02-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam 2.1G Nov 14 07:20 CF02-CM02-Zygote_dedup.sorted.bam
    -rw-r--r-- 1 sam sam 921K Nov 14 08:32 CF02-CM02-Zygote_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam 2.5G Nov 13 14:19 CF02-CM02-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.bam
    -rw-r--r-- 1 sam sam  319 Nov 13 14:19 CF02-CM02-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication_report.txt
    -rw-r--r-- 1 sam sam 3.4K Nov 13 14:19 CF02-CM02-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam 3.0G Nov 14 07:22 CF03-CM03-Zygote_dedup.sorted.bam
    -rw-r--r-- 1 sam sam 1.2M Nov 14 08:33 CF03-CM03-Zygote_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam 3.7G Nov 13 14:29 CF03-CM03-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.bam
    -rw-r--r-- 1 sam sam  318 Nov 13 14:29 CF03-CM03-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication_report.txt
    -rw-r--r-- 1 sam sam 3.4K Nov 13 14:29 CF03-CM03-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam 3.4G Nov 14 07:24 CF03-CM04-Larvae_dedup.sorted.bam
    -rw-r--r-- 1 sam sam 1.3M Nov 14 08:34 CF03-CM04-Larvae_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam 4.6G Nov 13 14:43 CF03-CM04-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.bam
    -rw-r--r-- 1 sam sam  320 Nov 13 14:43 CF03-CM04-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication_report.txt
    -rw-r--r-- 1 sam sam 3.4K Nov 13 14:43 CF03-CM04-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam 4.1G Nov 14 07:26 CF03-CM05-Larvae_dedup.sorted.bam
    -rw-r--r-- 1 sam sam 1.4M Nov 14 08:35 CF03-CM05-Larvae_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam 5.4G Nov 13 14:58 CF03-CM05-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.bam
    -rw-r--r-- 1 sam sam  318 Nov 13 14:58 CF03-CM05-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication_report.txt
    -rw-r--r-- 1 sam sam 3.4K Nov 13 14:58 CF03-CM05-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam 3.0G Nov 14 07:28 CF04-CM04-Zygote_dedup.sorted.bam
    -rw-r--r-- 1 sam sam 1.2M Nov 14 08:36 CF04-CM04-Zygote_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam 3.7G Nov 13 15:08 CF04-CM04-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.bam
    -rw-r--r-- 1 sam sam  319 Nov 13 15:08 CF04-CM04-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication_report.txt
    -rw-r--r-- 1 sam sam 3.4K Nov 13 15:08 CF04-CM04-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam 5.4G Nov 14 07:33 CF05-CM02-Larvae_dedup.sorted.bam
    -rw-r--r-- 1 sam sam 1.6M Nov 14 08:38 CF05-CM02-Larvae_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam 7.5G Nov 13 15:29 CF05-CM02-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.bam
    -rw-r--r-- 1 sam sam  320 Nov 13 15:29 CF05-CM02-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication_report.txt
    -rw-r--r-- 1 sam sam 3.4K Nov 13 15:30 CF05-CM02-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam 3.5G Nov 14 07:35 CF05-CM05-Zygote_dedup.sorted.bam
    -rw-r--r-- 1 sam sam 1.3M Nov 14 08:39 CF05-CM05-Zygote_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam 4.4G Nov 13 15:42 CF05-CM05-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.bam
    -rw-r--r-- 1 sam sam  319 Nov 13 15:42 CF05-CM05-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication_report.txt
    -rw-r--r-- 1 sam sam 3.4K Nov 13 15:42 CF05-CM05-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam 3.3G Nov 14 07:37 CF06-CM01-Zygote_dedup.sorted.bam
    -rw-r--r-- 1 sam sam 1.3M Nov 14 08:40 CF06-CM01-Zygote_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam 4.0G Nov 13 15:52 CF06-CM01-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.bam
    -rw-r--r-- 1 sam sam  318 Nov 13 15:52 CF06-CM01-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication_report.txt
    -rw-r--r-- 1 sam sam 3.4K Nov 13 15:53 CF06-CM01-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam 2.6G Nov 14 07:38 CF06-CM02-Larvae_dedup.sorted.bam
    -rw-r--r-- 1 sam sam 1.1M Nov 14 08:41 CF06-CM02-Larvae_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam 3.1G Nov 13 16:01 CF06-CM02-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.bam
    -rw-r--r-- 1 sam sam  318 Nov 13 16:01 CF06-CM02-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication_report.txt
    -rw-r--r-- 1 sam sam 3.4K Nov 13 16:01 CF06-CM02-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam 2.0G Nov 14 07:39 CF07-CM02-Zygote_dedup.sorted.bam
    -rw-r--r-- 1 sam sam 968K Nov 14 08:41 CF07-CM02-Zygote_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam 2.4G Nov 13 16:08 CF07-CM02-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.bam
    -rw-r--r-- 1 sam sam  318 Nov 13 16:08 CF07-CM02-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication_report.txt
    -rw-r--r-- 1 sam sam 3.4K Nov 13 16:08 CF07-CM02-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam 2.2G Nov 14 07:41 CF08-CM03-Zygote_dedup.sorted.bam
    -rw-r--r-- 1 sam sam 950K Nov 14 08:42 CF08-CM03-Zygote_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam 2.6G Nov 13 16:15 CF08-CM03-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.bam
    -rw-r--r-- 1 sam sam  319 Nov 13 16:15 CF08-CM03-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication_report.txt
    -rw-r--r-- 1 sam sam 3.4K Nov 13 16:15 CF08-CM03-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam  58M Nov 14 07:41 CF08-CM04-Larvae_dedup.sorted.bam
    -rw-r--r-- 1 sam sam 1.1M Nov 14 08:42 CF08-CM04-Larvae_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam  60M Nov 13 16:15 CF08-CM04-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.bam
    -rw-r--r-- 1 sam sam  312 Nov 13 16:15 CF08-CM04-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication_report.txt
    -rw-r--r-- 1 sam sam 3.4K Nov 13 16:15 CF08-CM04-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam  50M Nov 14 07:41 CF08-CM05-Larvae_dedup.sorted.bam
    -rw-r--r-- 1 sam sam 1.1M Nov 14 08:42 CF08-CM05-Larvae_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam  52M Nov 13 16:15 CF08-CM05-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.bam
    -rw-r--r-- 1 sam sam  314 Nov 13 16:15 CF08-CM05-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication_report.txt
    -rw-r--r-- 1 sam sam 3.4K Nov 13 16:15 CF08-CM05-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam  16K Nov 14 09:44 checksums.md5
    -rw-r--r-- 1 sam sam 2.8G Nov 14 07:42 EF01-EM01-Zygote_dedup.sorted.bam
    -rw-r--r-- 1 sam sam 1.2M Nov 14 08:43 EF01-EM01-Zygote_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam 3.5G Nov 13 16:25 EF01-EM01-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.bam
    -rw-r--r-- 1 sam sam  318 Nov 13 16:25 EF01-EM01-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication_report.txt
    -rw-r--r-- 1 sam sam 3.4K Nov 13 16:25 EF01-EM01-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam 3.0G Nov 14 07:44 EF02-EM02-Zygote_dedup.sorted.bam
    -rw-r--r-- 1 sam sam 1.2M Nov 14 08:44 EF02-EM02-Zygote_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam 3.7G Nov 13 16:35 EF02-EM02-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.bam
    -rw-r--r-- 1 sam sam  318 Nov 13 16:35 EF02-EM02-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication_report.txt
    -rw-r--r-- 1 sam sam 3.4K Nov 13 16:35 EF02-EM02-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam 2.0G Nov 14 07:45 EF03-EM03-Zygote_dedup.sorted.bam
    -rw-r--r-- 1 sam sam 897K Nov 14 08:45 EF03-EM03-Zygote_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam 2.3G Nov 13 16:41 EF03-EM03-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.bam
    -rw-r--r-- 1 sam sam  318 Nov 13 16:41 EF03-EM03-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication_report.txt
    -rw-r--r-- 1 sam sam 3.4K Nov 13 16:41 EF03-EM03-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam 4.1G Nov 14 07:47 EF03-EM04-Larvae_dedup.sorted.bam
    -rw-r--r-- 1 sam sam 1.4M Nov 14 08:46 EF03-EM04-Larvae_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam 5.1G Nov 13 16:55 EF03-EM04-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.bam
    -rw-r--r-- 1 sam sam  319 Nov 13 16:55 EF03-EM04-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication_report.txt
    -rw-r--r-- 1 sam sam 3.4K Nov 13 16:56 EF03-EM04-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam 4.1G Nov 14 07:50 EF03-EM05-Larvae_dedup.sorted.bam
    -rw-r--r-- 1 sam sam 1.5M Nov 14 08:47 EF03-EM05-Larvae_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam 5.4G Nov 13 17:10 EF03-EM05-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.bam
    -rw-r--r-- 1 sam sam  318 Nov 13 17:10 EF03-EM05-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication_report.txt
    -rw-r--r-- 1 sam sam 3.4K Nov 13 17:11 EF03-EM05-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam 1.1G Nov 14 07:50 EF04-EM04-Zygote_dedup.sorted.bam
    -rw-r--r-- 1 sam sam 600K Nov 14 08:47 EF04-EM04-Zygote_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam 1.2G Nov 13 17:14 EF04-EM04-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.bam
    -rw-r--r-- 1 sam sam  317 Nov 13 17:14 EF04-EM04-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication_report.txt
    -rw-r--r-- 1 sam sam 3.4K Nov 13 17:14 EF04-EM04-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam 4.9G Nov 14 07:55 EF04-EM05-Larvae_dedup.sorted.bam
    -rw-r--r-- 1 sam sam 1.5M Nov 14 08:49 EF04-EM05-Larvae_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam 6.8G Nov 13 17:33 EF04-EM05-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.bam
    -rw-r--r-- 1 sam sam  320 Nov 13 17:33 EF04-EM05-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication_report.txt
    -rw-r--r-- 1 sam sam 3.4K Nov 13 17:33 EF04-EM05-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam 3.1G Nov 14 07:57 EF05-EM01-Larvae_dedup.sorted.bam
    -rw-r--r-- 1 sam sam 1.2M Nov 14 08:50 EF05-EM01-Larvae_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam 4.1G Nov 13 17:45 EF05-EM01-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.bam
    -rw-r--r-- 1 sam sam  318 Nov 13 17:45 EF05-EM01-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication_report.txt
    -rw-r--r-- 1 sam sam 3.4K Nov 13 17:45 EF05-EM01-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam 702M Nov 14 07:57 EF05-EM05-Zygote_dedup.sorted.bam
    -rw-r--r-- 1 sam sam 567K Nov 14 08:50 EF05-EM05-Zygote_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam 756M Nov 13 17:47 EF05-EM05-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.bam
    -rw-r--r-- 1 sam sam  316 Nov 13 17:47 EF05-EM05-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication_report.txt
    -rw-r--r-- 1 sam sam 3.4K Nov 13 17:47 EF05-EM05-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam 5.0G Nov 14 08:02 EF05-EM06-Larvae_dedup.sorted.bam
    -rw-r--r-- 1 sam sam 1.6M Nov 14 08:52 EF05-EM06-Larvae_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam 7.0G Nov 13 18:07 EF05-EM06-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.bam
    -rw-r--r-- 1 sam sam  320 Nov 13 18:07 EF05-EM06-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication_report.txt
    -rw-r--r-- 1 sam sam 3.4K Nov 13 18:07 EF05-EM06-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam 4.9G Nov 14 08:06 EF06-EM01-Larvae_dedup.sorted.bam
    -rw-r--r-- 1 sam sam 1.6M Nov 14 08:53 EF06-EM01-Larvae_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam 6.8G Nov 13 18:26 EF06-EM01-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.bam
    -rw-r--r-- 1 sam sam  319 Nov 13 18:26 EF06-EM01-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication_report.txt
    -rw-r--r-- 1 sam sam 3.4K Nov 13 18:26 EF06-EM01-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam 4.0G Nov 14 08:09 EF06-EM02-Larvae_dedup.sorted.bam
    -rw-r--r-- 1 sam sam 1.4M Nov 14 08:55 EF06-EM02-Larvae_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam 5.2G Nov 13 18:41 EF06-EM02-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.bam
    -rw-r--r-- 1 sam sam  319 Nov 13 18:41 EF06-EM02-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication_report.txt
    -rw-r--r-- 1 sam sam 3.4K Nov 13 18:41 EF06-EM02-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam 1.2G Nov 14 08:09 EF06-EM06-Larvae_dedup.sorted.bam
    -rw-r--r-- 1 sam sam 574K Nov 14 08:55 EF06-EM06-Larvae_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam 1.3G Nov 13 18:48 EF06-EM06-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.bam
    -rw-r--r-- 1 sam sam  318 Nov 13 18:48 EF06-EM06-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication_report.txt
    -rw-r--r-- 1 sam sam 3.4K Nov 13 18:48 EF06-EM06-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam 1.5G Nov 14 08:10 EF07-EM01-Zygote_dedup.sorted.bam
    -rw-r--r-- 1 sam sam 730K Nov 14 08:55 EF07-EM01-Zygote_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam 1.7G Nov 13 18:53 EF07-EM01-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.bam
    -rw-r--r-- 1 sam sam  319 Nov 13 18:53 EF07-EM01-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication_report.txt
    -rw-r--r-- 1 sam sam 3.4K Nov 13 18:53 EF07-EM01-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam 3.8G Nov 14 08:12 EF07-EM03-Larvae_dedup.sorted.bam
    -rw-r--r-- 1 sam sam 1.4M Nov 14 08:57 EF07-EM03-Larvae_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam 5.2G Nov 13 19:08 EF07-EM03-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.bam
    -rw-r--r-- 1 sam sam  319 Nov 13 19:08 EF07-EM03-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication_report.txt
    -rw-r--r-- 1 sam sam 3.4K Nov 13 19:08 EF07-EM03-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam 4.3G Nov 14 08:17 EF08-EM03-Larvae_dedup.sorted.bam
    -rw-r--r-- 1 sam sam 1.4M Nov 14 08:58 EF08-EM03-Larvae_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam 6.0G Nov 13 19:25 EF08-EM03-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.bam
    -rw-r--r-- 1 sam sam  320 Nov 13 19:25 EF08-EM03-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication_report.txt
    -rw-r--r-- 1 sam sam 3.4K Nov 13 19:25 EF08-EM03-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication.stderr
    -rw-r--r-- 1 sam sam 3.8G Nov 14 08:19 EF08-EM04-Larvae_dedup.sorted.bam
    -rw-r--r-- 1 sam sam 1.3M Nov 14 08:59 EF08-EM04-Larvae_dedup.sorted.bam.bai
    -rw-r--r-- 1 sam sam 4.9G Nov 13 19:38 EF08-EM04-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.bam
    -rw-r--r-- 1 sam sam  319 Nov 13 19:38 EF08-EM04-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication_report.txt
    -rw-r--r-- 1 sam sam 3.4K Nov 13 19:39 EF08-EM04-Larvae_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplication.stderr
    drwxr-xr-x 2 sam sam 4.0K Nov 14 09:30 multiqc_data
    -rw-r--r-- 1 sam sam 1.1M Nov 14 09:30 multiqc_report.html

<div id="refs" class="references csl-bib-body hanging-indent">

<div id="ref-ewels2016" class="csl-entry">

Ewels, Philip, Måns Magnusson, Sverker Lundin, and Max Käller. 2016.
“MultiQC: Summarize Analysis Results for Multiple Tools and Samples in a
Single Report.” *Bioinformatics* 32 (19): 3047–48.
<https://doi.org/10.1093/bioinformatics/btw354>.

</div>

<div id="ref-krueger2011" class="csl-entry">

Krueger, Felix, and Simon R. Andrews. 2011. “Bismark: A Flexible Aligner
and Methylation Caller for Bisulfite-Seq Applications.” *Bioinformatics*
27 (11): 1571–72. <https://doi.org/10.1093/bioinformatics/btr167>.

</div>

</div>
