---
title: "02.20-bismark-methylation-extraction"
author: "Sam White"
date: "2024-11-26"
output: 
  bookdown::html_document2:
    theme: cosmo
    toc: true
    toc_float: true
    number_sections: true
    code_folding: show
    code_download: true
  github_document:
    toc: true
    number_sections: true
  html_document:
    theme: cosmo
    toc: true
    toc_float: true
    number_sections: true
    code_folding: show
    code_download: true
bibliography: references.bib
---
# Intro

This notebook is a continuation of using [Bismark](https://felixkrueger.github.io/Bismark/) [@krueger2011] to perform methylation analysis. This step will perform methylation extraction.

Results will be summarized by [MultiQC](https://github.com/MultiQC/MultiQC) [@ewels2016].

## Inputs

This notebook relies on deduplicated BAMs generated by [`02.10-bismark-deduplication.Rmd`](./02.10-bismark-deduplication.Rmd).

## Outputs

- `*.deduplicated.bedGraph.gz`: Bedgraph file with CpG loci and percent methylation.

- `*.deduplicated.bedGraph.gz.bismark.zero.cov`: Extension of `*deduplicated.bedGraph.gz` which includes two additional columns, count methylated and count unmethylated, in that order.

- `*.deduplicated.bismark.cov.gz`: 1-based counting version of `*.deduplicated.bedGraph.gz.bismark.zero.cov`. E.g. loci are at 49 49, instead of 48 49.

- `*.deduplicated.CpG_report.txt.gz`: Report on all CpGs. `<chromosome> <0-based start> <methylation status, +/-> <count_methyldated> <CG> <trinucleotide_context>`

- `*.deduplicated.cytosine_context_summary.txt`: Cytosine context summary. See file headers for info.

- `*.deduplicated.M-bias.txt`: Methylation bias table. See file headers for info.

- `*.deduplicated_splitting_report.txt`: Summary methylation report in a human-readable format.

- `CH*.deduplicated.txt.gz`: Strand-specific methylation status for each read in `CHH` or `CHG` context.

- `CpG*.deduplicated.txt.gz`: Strand-specific methylation status for each read in `CpG` context.

- `multiqc_report.html`: A summary report of the alignment results generated by [MultiQC](https://github.com/MultiQC/MultiQC), in HTML format.


Due to the large file sizes of most of the output files, these cannot be hosted in the [ceasmallr GitHub repo](https://github.com/sr320/ceasmallr). As such these files are available for download here:

- [https://gannet.fish.washington.edu/Atumefaciens/gitrepos/ceasmallr/output/02.20-bismark-methylation-extraction/](https://gannet.fish.washington.edu/Atumefaciens/gitrepos/ceasmallr/output/02.20-bismark-methylation-extraction/)



```{r setup, include=FALSE}
library(knitr)
knitr::opts_chunk$set(
  echo = TRUE,         # Display code chunks
  eval = FALSE,        # Evaluate code chunks
  warning = FALSE,     # Hide warnings
  message = FALSE,     # Hide messages
  comment = ""         # Prevents appending '##' to beginning of lines in code output
)
```

# Create a Bash variables file

This allows usage of Bash variables across R Markdown chunks.

```{r save-bash-variables-to-rvars-file, engine='bash', eval=TRUE}
{
echo "#### Assign Variables ####"
echo ""

echo "# Data directories"
echo 'export repo_dir=/home/shared/8TB_HDD_01/sam/gitrepos/ceasmallr'
echo 'export output_dir_top=${repo_dir}/output/02.20-bismark-methylation-extraction'
echo 'export deduplication_dir="${repo_dir}/output/02.10-bismark-deduplication"'
echo 'export genome_dir="${repo_dir}/data/Cvirginica_v300/"'
echo ""


echo "# Paths to programs"
echo 'export programs_dir="/home/shared"'
echo 'export bismark_dir="${programs_dir}/Bismark-0.24.0"'
echo 'export bowtie2_dir="${programs_dir}/bowtie2-2.4.4-linux-x86_64"'
echo 'export multiqc="/home/sam/programs/mambaforge/bin/multiqc"'
echo 'export samtools_dir="${programs_dir}/samtools-1.12"'
echo ""

echo "# Set number of CPUs to use"
echo 'export threads=40'


echo "# Print formatting"
echo 'export line="--------------------------------------------------------"'
echo ""
} > .bashvars

cat .bashvars
```

# Methylation Extraction

```{r methylation-extraction, engine='bash', eval=FALSE}
# Load bash variables into memory
source .bashvars

cd "${deduplication_dir}"

${bismark_dir}/bismark_methylation_extractor \
--bedGraph \
--counts \
--scaffolds \
--parallel 10 \
--output_dir ${output_dir_top} \
--samtools_path ${samtools_dir} \
--gzip \
--paired-end \
--comprehensive \
--merge_non_CpG \
--zero_based \
--buffer_size 75% \
--cytosine_report \
--genome_folder ${genome_dir} \
*deduplicated.bam
```

# Peek at output files
```{r head-files, engine='bash', eval=TRUE}
# Load bash variables into memory
source .bashvars

cd "${output_dir_top}"

head \
<(zcat CF01-CM01-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.bedGraph.gz) \
CF01-CM01-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.bedGraph.gz.bismark.zero.cov \
<(zcat CF01-CM01-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.bismark.cov.gz) \
<(zcat CF01-CM01-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.CpG_report.txt.gz) \
CF01-CM01-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.cytosine_context_summary.txt \
CF01-CM01-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated.M-bias.txt \
CF01-CM01-Zygote_R1_001.fastp-trim.20220827_bismark_bt2_pe.deduplicated_splitting_report.txt \
| column -t -s $'\t'
```

# List all outputs
```{r list-outputs, engine='bash', eval=TRUE}
# Load bash variables into memory
source .bashvars

cd "${output_dir_top}"

ls -lh
```

# Checksums
```{r generate-checksums, engine='bash', eval=FALSE}
# Load bash variables into memory
source .bashvars

cd "${output_dir_top}"

for file in *; do
  if [ "${file}" != "checksums.md5" ]; then
    md5sum "${file}" >> checksums.md5
  fi
done
```

# MultiQC
```{r multiqc, engine='bash', eval=FALSE}
# Load bash variables into memory
source .bashvars

cd "${output_dir_top}"

${multiqc} .
```



