02.20-bismark-methylation-extraction
================
Sam White
2024-11-26

- <a href="#1-intro" id="toc-1-intro">1 Intro</a>
  - <a href="#11-inputs" id="toc-11-inputs">1.1 Inputs</a>
  - <a href="#12-outputs" id="toc-12-outputs">1.2 Outputs</a>
- <a href="#2-create-a-bash-variables-file"
  id="toc-2-create-a-bash-variables-file">2 Create a Bash variables
  file</a>
- <a href="#3-methylation-extraction" id="toc-3-methylation-extraction">3
  Methylation Extraction</a>
- <a href="#4-cytosine-coverage" id="toc-4-cytosine-coverage">4 Cytosine
  coverage</a>
- <a href="#5-peek-at-output-files" id="toc-5-peek-at-output-files">5 Peek
  at output files</a>
- <a href="#6-list-all-outputs" id="toc-6-list-all-outputs">6 List all
  outputs</a>
- <a href="#7-checksums" id="toc-7-checksums">7 Checksums</a>
- <a href="#8-multiqc" id="toc-8-multiqc">8 MultiQC</a>

# 1 Intro

This notebook is a continuation of using
[Bismark](https://felixkrueger.github.io/Bismark/) (Krueger and Andrews
2011) to perform methylation analysis. This step will perform
methylation extraction.

Results will be summarized by
[MultiQC](https://github.com/MultiQC/MultiQC) (Ewels et al. 2016).

## 1.1 Inputs

This notebook relies on deduplicated BAMs generated by
[`02.10-bismark-deduplication.Rmd`](./02.10-bismark-deduplication.Rmd).

## 1.2 Outputs

- `*.CpG_report.merged_CpG_evidence.cov.gz`: CpG position and counts
  merged from all strands. Generated by `coverage2cytosine`.

- `*.deduplicated.bedGraph.gz`: Bedgraph file with CpG loci and percent
  methylation.

- `*.deduplicated.bedGraph.gz.bismark.zero.cov`: Extension of
  `*deduplicated.bedGraph.gz` which includes two additional columns,
  count methylated and count unmethylated, in that order.

- `*.deduplicated.bismark.cov.gz`: 1-based counting version of
  `*.deduplicated.bedGraph.gz.bismark.zero.cov`. E.g. loci are at 49 49,
  instead of 48 49.

- `*.deduplicated.CpG_report.txt.gz`: Report on all CpGs.
  `<chromosome> <0-based start> <methylation status, +/-> <count_methyldated> <CG> <trinucleotide_context>`

- `*.deduplicated.cytosine_context_summary.txt`: Cytosine context
  summary. See file headers for info.

- `*.deduplicated.M-bias.txt`: Methylation bias table. See file headers
  for info.

- `*.deduplicated_splitting_report.txt`: Summary methylation report in a
  human-readable format.

- `CH*.deduplicated.txt.gz`: Strand-specific methylation status for each
  read in `CHH` or `CHG` context.

- `CpG*.deduplicated.txt.gz`: Strand-specific methylation status for
  each read in `CpG` context.

- `multiqc_report.html`: A summary report of the alignment results
  generated by [MultiQC](https://github.com/MultiQC/MultiQC), in HTML
  format.

Due to the large file sizes of most of the output files, these cannot be
hosted in the [ceasmallr GitHub
repo](https://github.com/sr320/ceasmallr). As such these files are
available for download here:

- <https://gannet.fish.washington.edu/gitrepos/ceasmallr/output/02.20-bismark-methylation-extraction/>

# 2 Create a Bash variables file

This allows usage of Bash variables across R Markdown chunks.

``` bash
{
echo "#### Assign Variables ####"
echo ""

echo "# Data directories"
echo 'export repo_dir=/home/shared/8TB_HDD_01/sam/gitrepos/ceasmallr'
echo 'export output_dir_top=${repo_dir}/output/02.20-bismark-methylation-extraction'
echo 'export deduplication_dir="${repo_dir}/output/02.10-bismark-deduplication"'
echo 'export genome_dir="${repo_dir}/data/Cvirginica_v300/"'
echo ""


echo "# Paths to programs"
echo 'export programs_dir="/home/shared"'
echo 'export bismark_dir="${programs_dir}/Bismark-0.24.0"'
echo 'export bowtie2_dir="${programs_dir}/bowtie2-2.4.4-linux-x86_64"'
echo 'export multiqc="/home/sam/programs/mambaforge/bin/multiqc"'
echo 'export samtools_dir="${programs_dir}/samtools-1.12"'
echo ""

echo "# Set number of CPUs to use"
echo 'export threads=40'


echo "# Print formatting"
echo 'export line="--------------------------------------------------------"'
echo ""
} > .bashvars

cat .bashvars
```

    #### Assign Variables ####

    # Data directories
    export repo_dir=/home/shared/8TB_HDD_01/sam/gitrepos/ceasmallr
    export output_dir_top=${repo_dir}/output/02.20-bismark-methylation-extraction
    export deduplication_dir="${repo_dir}/output/02.10-bismark-deduplication"
    export genome_dir="${repo_dir}/data/Cvirginica_v300/"

    # Paths to programs
    export programs_dir="/home/shared"
    export bismark_dir="${programs_dir}/Bismark-0.24.0"
    export bowtie2_dir="${programs_dir}/bowtie2-2.4.4-linux-x86_64"
    export multiqc="/home/sam/programs/mambaforge/bin/multiqc"
    export samtools_dir="${programs_dir}/samtools-1.12"

    # Set number of CPUs to use
    export threads=40
    # Print formatting
    export line="--------------------------------------------------------"

# 3 Methylation Extraction

``` bash
# Load bash variables into memory
source .bashvars

cd "${deduplication_dir}"

${bismark_dir}/bismark_methylation_extractor \
--bedGraph \
--counts \
--scaffolds \
--parallel 10 \
--output_dir ${output_dir_top} \
--samtools_path ${samtools_dir} \
--gzip \
--paired-end \
--comprehensive \
--merge_non_CpG \
--zero_based \
--buffer_size 75% \
--cytosine_report \
--genome_folder ${genome_dir} \
*deduplicated.bam
```

# 4 Cytosine coverage

``` bash
# Load bash variables into memory
source .bashvars

cd "${output_dir_top}"

find *deduplicated.bismark.cov.gz \
| xargs basename -s _bismark_bt2_pe.deduplicated.bismark.cov.gz \
| xargs -I{} ${bismark_dir}/coverage2cytosine \
--genome_folder ${genome_dir} \
-o {} \
--merge_CpG \
--zero_based \
--gzip \
{}_bismark_bt2_pe.deduplicated.bismark.cov.gz
```

# 5 Peek at output files

Glance at output file formats, using CF01-CM01-Zygote files as example.

``` bash
# Load bash variables into memory
source .bashvars

cd "${output_dir_top}"

for file in *CF01-CM01-Zygote*; do
  if [[ -f "$file" ]]; then
    if [[ "$file" == *.gz ]]; then
      echo "<===== $file ====>"
      echo ""
      zcat "$file" | head | column -t -s $'\t'
      echo ""
      echo "${line}"
      echo ""
    else
      echo "<===== $file ====>"
      echo ""
      echo "$file"
      head "$file" | column -t -s $'\t'
      echo ""
      echo "${line}"
      echo ""
    fi
  fi
done
```

    <===== CF01-CM01-Zygote_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.bedGraph.gz ====>

    track type=bedGraph
    NC_007175.2          48   49   2.08333333333333
    NC_007175.2          49   50   0
    NC_007175.2          50   51   1.92307692307692
    NC_007175.2          51   52   0
    NC_007175.2          87   88   0
    NC_007175.2          88   89   0
    NC_007175.2          146  147  0
    NC_007175.2          147  148  1.75438596491228
    NC_007175.2          148  149  0

    --------------------------------------------------------

    <===== CF01-CM01-Zygote_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.bedGraph.gz.bismark.zero.cov ====>

    CF01-CM01-Zygote_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.bedGraph.gz.bismark.zero.cov
    NC_007175.2  48   49   2.08333333333333  1  47
    NC_007175.2  49   50   0                 0  2
    NC_007175.2  50   51   1.92307692307692  1  51
    NC_007175.2  51   52   0                 0  1
    NC_007175.2  87   88   0                 0  104
    NC_007175.2  88   89   0                 0  33
    NC_007175.2  146  147  0                 0  116
    NC_007175.2  147  148  1.75438596491228  1  56
    NC_007175.2  148  149  0                 0  1
    NC_007175.2  192  193  0                 0  51

    --------------------------------------------------------

    <===== CF01-CM01-Zygote_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.bismark.cov.gz ====>

    NC_007175.2  49   49   2.08333333333333  1  47
    NC_007175.2  50   50   0                 0  2
    NC_007175.2  51   51   1.92307692307692  1  51
    NC_007175.2  52   52   0                 0  1
    NC_007175.2  88   88   0                 0  104
    NC_007175.2  89   89   0                 0  33
    NC_007175.2  147  147  0                 0  116
    NC_007175.2  148  148  1.75438596491228  1  56
    NC_007175.2  149  149  0                 0  1
    NC_007175.2  193  193  0                 0  51

    --------------------------------------------------------

    <===== CF01-CM01-Zygote_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.CpG_report.txt.gz ====>

    NC_007175.2  48   +  1  47   CG  CGC
    NC_007175.2  49   -  0  2    CG  CGA
    NC_007175.2  50   +  1  51   CG  CGG
    NC_007175.2  51   -  0  1    CG  CGC
    NC_007175.2  87   +  0  104  CG  CGT
    NC_007175.2  88   -  0  33   CG  CGT
    NC_007175.2  146  +  0  116  CG  CGG
    NC_007175.2  147  -  1  56   CG  CGA
    NC_007175.2  192  +  0  51   CG  CGC
    NC_007175.2  193  -  0  39   CG  CGA

    --------------------------------------------------------

    <===== CF01-CM01-Zygote_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.cytosine_context_summary.txt ====>

    CF01-CM01-Zygote_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.cytosine_context_summary.txt
    upstream  C-context  full context  count methylated  count unmethylated  percent methylation
    A         CAA        ACAA          66                446                 12.89
    C         CAA        CCAA          66                479                 12.11
    G         CAA        GCAA          19                198                 8.76
    T         CAA        TCAA          59                526                 10.09
    A         CAC        ACAC          16                118                 11.94
    C         CAC        CCAC          7                 69                  9.21
    G         CAC        GCAC          9                 38                  19.15
    T         CAC        TCAC          15                120                 11.11
    A         CAG        ACAG          142               1235                10.31

    --------------------------------------------------------

    <===== CF01-CM01-Zygote_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.M-bias.txt ====>

    CF01-CM01-Zygote_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.M-bias.txt
    CpG context (R1)
    ================
    position          count methylated  count unmethylated  % methylation  coverage
    1                 9383              45046               17.24          54429
    2                 9270              45432               16.95          54702
    3                 9184              45622               16.76          54806
    4                 9457              45750               17.13          55207
    5                 10083             48122               17.32          58205
    6                 10548             48157               17.97          58705
    7                 10617             48528               17.95          59145

    --------------------------------------------------------

    <===== CF01-CM01-Zygote_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated_splitting_report.txt ====>

    CF01-CM01-Zygote_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated_splitting_report.txt
    CF01-CM01-Zygote_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.bam
    Parameters used to extract methylation information:
    Bismark Extractor Version: v0.24.0
    Bismark result file: paired-end (SAM format)
    Output specified: comprehensive
    No overlapping methylation calls specified
    Methylation in CHG and CHH context will be merged into "non-CpG context" output

    --------------------------------------------------------

    <===== CF01-CM01-Zygote_R1_001.fastp-trim.REPAIRED.CpG_report.merged_CpG_evidence.cov.gz ====>

    NC_007175.2  48   50   2.000000  1  49
    NC_007175.2  50   52   1.886792  1  52
    NC_007175.2  87   89   0.000000  0  137
    NC_007175.2  146  148  0.578035  1  172
    NC_007175.2  192  194  0.000000  0  90
    NC_007175.2  245  247  2.898551  2  67
    NC_007175.2  256  258  4.166667  5  115
    NC_007175.2  263  265  0.000000  0  155
    NC_007175.2  265  267  1.142857  2  173
    NC_007175.2  331  333  0.546448  1  182

    --------------------------------------------------------

    <===== CF01-CM01-Zygote_R1_001.fastp-trim.REPAIRED.CpG_report.txt.gz ====>

    NC_007175.2  48   +  1  47   CG  CGC
    NC_007175.2  49   -  0  2    CG  CGA
    NC_007175.2  50   +  1  51   CG  CGG
    NC_007175.2  51   -  0  1    CG  CGC
    NC_007175.2  87   +  0  104  CG  CGT
    NC_007175.2  88   -  0  33   CG  CGT
    NC_007175.2  146  +  0  116  CG  CGG
    NC_007175.2  147  -  1  56   CG  CGA
    NC_007175.2  192  +  0  51   CG  CGC
    NC_007175.2  193  -  0  39   CG  CGA

    --------------------------------------------------------

    <===== CF01-CM01-Zygote_R1_001.fastp-trim.REPAIRED.cytosine_context_summary.txt ====>

    CF01-CM01-Zygote_R1_001.fastp-trim.REPAIRED.cytosine_context_summary.txt
    upstream  C-context  full context  count methylated  count unmethylated  percent methylation
    A         CAA        ACAA          66                446                 12.89
    C         CAA        CCAA          66                479                 12.11
    G         CAA        GCAA          19                198                 8.76
    T         CAA        TCAA          59                526                 10.09
    A         CAC        ACAC          16                118                 11.94
    C         CAC        CCAC          7                 69                  9.21
    G         CAC        GCAC          9                 38                  19.15
    T         CAC        TCAC          15                120                 11.11
    A         CAG        ACAG          142               1235                10.31

    --------------------------------------------------------

    <===== CpG_context_CF01-CM01-Zygote_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.txt.gz ====>

    Bismark methylation extractor version v0.24.0
    GWNJ-1012:512:GW210315000:4:1101:21884:1094_1:N:0:TTATAACC+TCGATATC  -  NC_035785.1  24407680  z
    GWNJ-1012:512:GW210315000:4:1101:11776:1219_1:N:0:TTATAACC+TCGATATC  -  NC_035780.1  32653438  z
    GWNJ-1012:512:GW210315000:4:1101:20699:1141_1:N:0:TTATAACC+TCGATATC  -  NC_035788.1  57095043  z
    GWNJ-1012:512:GW210315000:4:1101:20699:1141_1:N:0:TTATAACC+TCGATATC  -  NC_035788.1  57095087  z
    GWNJ-1012:512:GW210315000:4:1101:20699:1141_1:N:0:TTATAACC+TCGATATC  -  NC_035788.1  57095119  z
    GWNJ-1012:512:GW210315000:4:1101:20699:1141_1:N:0:TTATAACC+TCGATATC  -  NC_035788.1  57095138  z
    GWNJ-1012:512:GW210315000:4:1101:9372:1595_1:N:0:TTATAACC+TCGATATC   +  NC_035781.1  2018694   Z
    GWNJ-1012:512:GW210315000:4:1101:15402:1642_1:N:0:TTATAACC+TCGATATC  +  NC_035783.1  49860452  Z
    GWNJ-1012:512:GW210315000:4:1101:9372:1595_1:N:0:TTATAACC+TCGATATC   +  NC_035781.1  2018682   Z

    --------------------------------------------------------

    <===== Non_CpG_context_CF01-CM01-Zygote_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.txt.gz ====>

    Bismark methylation extractor version v0.24.0
    GWNJ-1012:512:GW210315000:4:1101:11776:1219_1:N:0:TTATAACC+TCGATATC  -  NC_035780.1  32653467  x
    GWNJ-1012:512:GW210315000:4:1101:11776:1219_1:N:0:TTATAACC+TCGATATC  -  NC_035780.1  32653466  h
    GWNJ-1012:512:GW210315000:4:1101:11776:1219_1:N:0:TTATAACC+TCGATATC  -  NC_035780.1  32653465  h
    GWNJ-1012:512:GW210315000:4:1101:11776:1219_1:N:0:TTATAACC+TCGATATC  -  NC_035780.1  32653464  h
    GWNJ-1012:512:GW210315000:4:1101:11776:1219_1:N:0:TTATAACC+TCGATATC  -  NC_035780.1  32653461  x
    GWNJ-1012:512:GW210315000:4:1101:11776:1219_1:N:0:TTATAACC+TCGATATC  -  NC_035780.1  32653455  h
    GWNJ-1012:512:GW210315000:4:1101:11776:1219_1:N:0:TTATAACC+TCGATATC  -  NC_035780.1  32653454  h
    GWNJ-1012:512:GW210315000:4:1101:11776:1219_1:N:0:TTATAACC+TCGATATC  -  NC_035780.1  32653450  h
    GWNJ-1012:512:GW210315000:4:1101:20699:1141_1:N:0:TTATAACC+TCGATATC  +  NC_035788.1  57095003  X

    --------------------------------------------------------

# 6 List all outputs

``` bash
# Load bash variables into memory
source .bashvars

cd "${output_dir_top}"

ls -lh
```

    total 93G
    -rw-r--r-- 1 sam sam  21M Nov 26 11:17 CF01-CM01-Zygote_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.bedGraph.gz
    -rw-r--r-- 1 sam sam 124M Nov 26 11:17 CF01-CM01-Zygote_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.bedGraph.gz.bismark.zero.cov
    -rw-r--r-- 1 sam sam  18M Nov 26 11:17 CF01-CM01-Zygote_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.bismark.cov.gz
    -rw-r--r-- 1 sam sam 107M Nov 26 11:27 CF01-CM01-Zygote_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.7K Dec  4 12:17 CF01-CM01-Zygote_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  20K Dec  4 12:17 CF01-CM01-Zygote_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.M-bias.txt
    -rw-r--r-- 1 sam sam  926 Dec  4 12:17 CF01-CM01-Zygote_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated_splitting_report.txt
    -rw-r--r-- 1 sam sam  21M Dec  4 12:17 CF01-CM01-Zygote_R1_001.fastp-trim.REPAIRED.CpG_report.merged_CpG_evidence.cov.gz
    -rw-r--r-- 1 sam sam 107M Nov 27 07:59 CF01-CM01-Zygote_R1_001.fastp-trim.REPAIRED.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.7K Dec  4 12:17 CF01-CM01-Zygote_R1_001.fastp-trim.REPAIRED.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  90M Nov 26 11:43 CF01-CM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz
    -rw-r--r-- 1 sam sam 551M Nov 26 11:43 CF01-CM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz.bismark.zero.cov
    -rw-r--r-- 1 sam sam  89M Nov 26 11:43 CF01-CM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bismark.cov.gz
    -rw-r--r-- 1 sam sam 126M Nov 26 11:54 CF01-CM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 CF01-CM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  23K Dec  4 12:17 CF01-CM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.M-bias.txt
    -rw-r--r-- 1 sam sam  928 Dec  4 12:17 CF01-CM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated_splitting_report.txt
    -rw-r--r-- 1 sam sam  72M Dec  4 12:17 CF01-CM02-Larvae_R1_001.fastp-trim.CpG_report.merged_CpG_evidence.cov.gz
    -rw-r--r-- 1 sam sam 126M Nov 27 08:11 CF01-CM02-Larvae_R1_001.fastp-trim.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 CF01-CM02-Larvae_R1_001.fastp-trim.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  44M Nov 26 12:00 CF02-CM02-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz
    -rw-r--r-- 1 sam sam 263M Nov 26 12:00 CF02-CM02-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz.bismark.zero.cov
    -rw-r--r-- 1 sam sam  41M Nov 26 12:00 CF02-CM02-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bismark.cov.gz
    -rw-r--r-- 1 sam sam 113M Nov 26 12:12 CF02-CM02-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 CF02-CM02-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  21K Dec  4 12:17 CF02-CM02-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.M-bias.txt
    -rw-r--r-- 1 sam sam  926 Dec  4 12:17 CF02-CM02-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated_splitting_report.txt
    -rw-r--r-- 1 sam sam  41M Dec  4 12:17 CF02-CM02-Zygote_R1_001.fastp-trim.CpG_report.merged_CpG_evidence.cov.gz
    -rw-r--r-- 1 sam sam 113M Nov 27 08:22 CF02-CM02-Zygote_R1_001.fastp-trim.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 CF02-CM02-Zygote_R1_001.fastp-trim.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  65M Nov 26 12:20 CF03-CM03-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz
    -rw-r--r-- 1 sam sam 395M Nov 26 12:20 CF03-CM03-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz.bismark.zero.cov
    -rw-r--r-- 1 sam sam  61M Nov 26 12:20 CF03-CM03-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bismark.cov.gz
    -rw-r--r-- 1 sam sam 118M Nov 26 12:32 CF03-CM03-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 CF03-CM03-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  22K Dec  4 12:17 CF03-CM03-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.M-bias.txt
    -rw-r--r-- 1 sam sam  925 Dec  4 12:17 CF03-CM03-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated_splitting_report.txt
    -rw-r--r-- 1 sam sam  57M Dec  4 12:17 CF03-CM03-Zygote_R1_001.fastp-trim.CpG_report.merged_CpG_evidence.cov.gz
    -rw-r--r-- 1 sam sam 118M Nov 27 08:33 CF03-CM03-Zygote_R1_001.fastp-trim.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 CF03-CM03-Zygote_R1_001.fastp-trim.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  85M Nov 26 12:45 CF03-CM04-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz
    -rw-r--r-- 1 sam sam 528M Nov 26 12:45 CF03-CM04-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz.bismark.zero.cov
    -rw-r--r-- 1 sam sam  83M Nov 26 12:45 CF03-CM04-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bismark.cov.gz
    -rw-r--r-- 1 sam sam 123M Nov 26 12:57 CF03-CM04-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 CF03-CM04-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  23K Dec  4 12:17 CF03-CM04-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.M-bias.txt
    -rw-r--r-- 1 sam sam  927 Dec  4 12:17 CF03-CM04-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated_splitting_report.txt
    -rw-r--r-- 1 sam sam  69M Dec  4 12:17 CF03-CM04-Larvae_R1_001.fastp-trim.CpG_report.merged_CpG_evidence.cov.gz
    -rw-r--r-- 1 sam sam 123M Nov 27 08:45 CF03-CM04-Larvae_R1_001.fastp-trim.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 CF03-CM04-Larvae_R1_001.fastp-trim.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  90M Nov 26 13:12 CF03-CM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz
    -rw-r--r-- 1 sam sam 551M Nov 26 13:12 CF03-CM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz.bismark.zero.cov
    -rw-r--r-- 1 sam sam  88M Nov 26 13:12 CF03-CM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bismark.cov.gz
    -rw-r--r-- 1 sam sam 125M Nov 26 13:23 CF03-CM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 CF03-CM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  23K Dec  4 12:17 CF03-CM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.M-bias.txt
    -rw-r--r-- 1 sam sam  928 Dec  4 12:17 CF03-CM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated_splitting_report.txt
    -rw-r--r-- 1 sam sam  71M Dec  4 12:17 CF03-CM05-Larvae_R1_001.fastp-trim.CpG_report.merged_CpG_evidence.cov.gz
    -rw-r--r-- 1 sam sam 125M Nov 27 08:57 CF03-CM05-Larvae_R1_001.fastp-trim.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 CF03-CM05-Larvae_R1_001.fastp-trim.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  74M Nov 26 13:33 CF04-CM04-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz
    -rw-r--r-- 1 sam sam 458M Nov 26 13:33 CF04-CM04-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz.bismark.zero.cov
    -rw-r--r-- 1 sam sam  71M Nov 26 13:33 CF04-CM04-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bismark.cov.gz
    -rw-r--r-- 1 sam sam 120M Nov 26 13:44 CF04-CM04-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 CF04-CM04-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  22K Dec  4 12:17 CF04-CM04-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.M-bias.txt
    -rw-r--r-- 1 sam sam  925 Dec  4 12:17 CF04-CM04-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated_splitting_report.txt
    -rw-r--r-- 1 sam sam  62M Dec  4 12:17 CF04-CM04-Zygote_R1_001.fastp-trim.CpG_report.merged_CpG_evidence.cov.gz
    -rw-r--r-- 1 sam sam 120M Nov 27 09:09 CF04-CM04-Zygote_R1_001.fastp-trim.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 CF04-CM04-Zygote_R1_001.fastp-trim.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  97M Nov 26 14:05 CF05-CM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz
    -rw-r--r-- 1 sam sam 599M Nov 26 14:05 CF05-CM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz.bismark.zero.cov
    -rw-r--r-- 1 sam sam  98M Nov 26 14:05 CF05-CM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bismark.cov.gz
    -rw-r--r-- 1 sam sam 128M Nov 26 14:16 CF05-CM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 CF05-CM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  23K Dec  4 12:17 CF05-CM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.M-bias.txt
    -rw-r--r-- 1 sam sam  928 Dec  4 12:17 CF05-CM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated_splitting_report.txt
    -rw-r--r-- 1 sam sam  78M Dec  4 12:17 CF05-CM02-Larvae_R1_001.fastp-trim.CpG_report.merged_CpG_evidence.cov.gz
    -rw-r--r-- 1 sam sam 128M Nov 27 09:20 CF05-CM02-Larvae_R1_001.fastp-trim.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 CF05-CM02-Larvae_R1_001.fastp-trim.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  82M Nov 26 14:28 CF05-CM05-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz
    -rw-r--r-- 1 sam sam 506M Nov 26 14:28 CF05-CM05-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz.bismark.zero.cov
    -rw-r--r-- 1 sam sam  80M Nov 26 14:28 CF05-CM05-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bismark.cov.gz
    -rw-r--r-- 1 sam sam 122M Nov 26 14:39 CF05-CM05-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 CF05-CM05-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  23K Dec  4 12:17 CF05-CM05-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.M-bias.txt
    -rw-r--r-- 1 sam sam  927 Dec  4 12:17 CF05-CM05-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated_splitting_report.txt
    -rw-r--r-- 1 sam sam  67M Dec  4 12:17 CF05-CM05-Zygote_R1_001.fastp-trim.CpG_report.merged_CpG_evidence.cov.gz
    -rw-r--r-- 1 sam sam 122M Nov 27 09:32 CF05-CM05-Zygote_R1_001.fastp-trim.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 CF05-CM05-Zygote_R1_001.fastp-trim.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  74M Nov 26 14:50 CF06-CM01-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz
    -rw-r--r-- 1 sam sam 452M Nov 26 14:50 CF06-CM01-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz.bismark.zero.cov
    -rw-r--r-- 1 sam sam  70M Nov 26 14:50 CF06-CM01-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bismark.cov.gz
    -rw-r--r-- 1 sam sam 120M Nov 26 15:01 CF06-CM01-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 CF06-CM01-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  22K Dec  4 12:17 CF06-CM01-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.M-bias.txt
    -rw-r--r-- 1 sam sam  926 Dec  4 12:17 CF06-CM01-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated_splitting_report.txt
    -rw-r--r-- 1 sam sam  63M Dec  4 12:17 CF06-CM01-Zygote_R1_001.fastp-trim.CpG_report.merged_CpG_evidence.cov.gz
    -rw-r--r-- 1 sam sam 120M Nov 27 09:44 CF06-CM01-Zygote_R1_001.fastp-trim.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 CF06-CM01-Zygote_R1_001.fastp-trim.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  73M Nov 26 15:09 CF06-CM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz
    -rw-r--r-- 1 sam sam 444M Nov 26 15:09 CF06-CM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz.bismark.zero.cov
    -rw-r--r-- 1 sam sam  68M Nov 26 15:09 CF06-CM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bismark.cov.gz
    -rw-r--r-- 1 sam sam 119M Nov 26 15:20 CF06-CM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 CF06-CM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  23K Dec  4 12:17 CF06-CM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.M-bias.txt
    -rw-r--r-- 1 sam sam  926 Dec  4 12:17 CF06-CM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated_splitting_report.txt
    -rw-r--r-- 1 sam sam  62M Dec  4 12:17 CF06-CM02-Larvae_R1_001.fastp-trim.CpG_report.merged_CpG_evidence.cov.gz
    -rw-r--r-- 1 sam sam 119M Nov 27 09:56 CF06-CM02-Larvae_R1_001.fastp-trim.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 CF06-CM02-Larvae_R1_001.fastp-trim.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  62M Nov 26 15:27 CF07-CM02-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz
    -rw-r--r-- 1 sam sam 383M Nov 26 15:27 CF07-CM02-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz.bismark.zero.cov
    -rw-r--r-- 1 sam sam  57M Nov 26 15:27 CF07-CM02-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bismark.cov.gz
    -rw-r--r-- 1 sam sam 116M Nov 26 15:38 CF07-CM02-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 CF07-CM02-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  22K Dec  4 12:17 CF07-CM02-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.M-bias.txt
    -rw-r--r-- 1 sam sam  924 Dec  4 12:17 CF07-CM02-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated_splitting_report.txt
    -rw-r--r-- 1 sam sam  54M Dec  4 12:17 CF07-CM02-Zygote_R1_001.fastp-trim.CpG_report.merged_CpG_evidence.cov.gz
    -rw-r--r-- 1 sam sam 116M Nov 27 10:07 CF07-CM02-Zygote_R1_001.fastp-trim.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 CF07-CM02-Zygote_R1_001.fastp-trim.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  48M Nov 26 15:44 CF08-CM03-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz
    -rw-r--r-- 1 sam sam 287M Nov 26 15:44 CF08-CM03-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz.bismark.zero.cov
    -rw-r--r-- 1 sam sam  44M Nov 26 15:44 CF08-CM03-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bismark.cov.gz
    -rw-r--r-- 1 sam sam 114M Nov 26 15:55 CF08-CM03-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 CF08-CM03-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  22K Dec  4 12:17 CF08-CM03-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.M-bias.txt
    -rw-r--r-- 1 sam sam  925 Dec  4 12:17 CF08-CM03-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated_splitting_report.txt
    -rw-r--r-- 1 sam sam  44M Dec  4 12:17 CF08-CM03-Zygote_R1_001.fastp-trim.CpG_report.merged_CpG_evidence.cov.gz
    -rw-r--r-- 1 sam sam 114M Nov 27 10:18 CF08-CM03-Zygote_R1_001.fastp-trim.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 CF08-CM03-Zygote_R1_001.fastp-trim.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam 3.8M Nov 26 15:55 CF08-CM04-Larvae_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.bedGraph.gz
    -rw-r--r-- 1 sam sam  22M Nov 26 15:55 CF08-CM04-Larvae_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.bedGraph.gz.bismark.zero.cov
    -rw-r--r-- 1 sam sam 3.0M Nov 26 15:55 CF08-CM04-Larvae_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.bismark.cov.gz
    -rw-r--r-- 1 sam sam 103M Nov 26 16:05 CF08-CM04-Larvae_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.6K Dec  4 12:17 CF08-CM04-Larvae_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  19K Dec  4 12:17 CF08-CM04-Larvae_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.M-bias.txt
    -rw-r--r-- 1 sam sam  919 Dec  4 12:17 CF08-CM04-Larvae_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated_splitting_report.txt
    -rw-r--r-- 1 sam sam 3.8M Dec  4 12:17 CF08-CM04-Larvae_R1_001.fastp-trim.REPAIRED.CpG_report.merged_CpG_evidence.cov.gz
    -rw-r--r-- 1 sam sam 103M Nov 27 10:28 CF08-CM04-Larvae_R1_001.fastp-trim.REPAIRED.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.6K Dec  4 12:17 CF08-CM04-Larvae_R1_001.fastp-trim.REPAIRED.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam 3.7M Nov 26 16:05 CF08-CM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz
    -rw-r--r-- 1 sam sam  22M Nov 26 16:05 CF08-CM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz.bismark.zero.cov
    -rw-r--r-- 1 sam sam 3.0M Nov 26 16:05 CF08-CM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bismark.cov.gz
    -rw-r--r-- 1 sam sam 103M Nov 26 16:15 CF08-CM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.6K Dec  4 12:17 CF08-CM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  19K Dec  4 12:17 CF08-CM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.M-bias.txt
    -rw-r--r-- 1 sam sam  910 Dec  4 12:17 CF08-CM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated_splitting_report.txt
    -rw-r--r-- 1 sam sam 3.8M Dec  4 12:17 CF08-CM05-Larvae_R1_001.fastp-trim.CpG_report.merged_CpG_evidence.cov.gz
    -rw-r--r-- 1 sam sam 103M Nov 27 10:38 CF08-CM05-Larvae_R1_001.fastp-trim.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.6K Dec  4 12:17 CF08-CM05-Larvae_R1_001.fastp-trim.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  43K Dec  4 12:17 checksums.md5
    -rw-r--r-- 1 sam sam  45M Nov 26 11:16 CpG_context_CF01-CM01-Zygote_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 725M Nov 26 11:37 CpG_context_CF01-CM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 191M Nov 26 11:59 CpG_context_CF02-CM02-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 323M Nov 26 12:18 CpG_context_CF03-CM03-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 557M Nov 26 12:41 CpG_context_CF03-CM04-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 664M Nov 26 13:07 CpG_context_CF03-CM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 387M Nov 26 13:30 CpG_context_CF04-CM04-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 943M Nov 26 13:58 CpG_context_CF05-CM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 496M Nov 26 14:25 CpG_context_CF05-CM05-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 390M Nov 26 14:47 CpG_context_CF06-CM01-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 344M Nov 26 15:07 CpG_context_CF06-CM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 240M Nov 26 15:25 CpG_context_CF07-CM02-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 215M Nov 26 15:42 CpG_context_CF08-CM03-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 5.7M Nov 26 15:55 CpG_context_CF08-CM04-Larvae_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 5.8M Nov 26 16:05 CpG_context_CF08-CM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 368M Nov 26 16:21 CpG_context_EF01-EM01-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 427M Nov 26 16:42 CpG_context_EF02-EM02-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 211M Nov 26 17:01 CpG_context_EF03-EM03-Zygote_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 607M Nov 26 17:23 CpG_context_EF03-EM04-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 625M Nov 26 17:49 CpG_context_EF03-EM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 123M Nov 26 18:08 CpG_context_EF04-EM04-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 857M Nov 26 18:32 CpG_context_EF04-EM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 514M Nov 26 18:58 CpG_context_EF05-EM01-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam  68M Nov 26 19:14 CpG_context_EF05-EM05-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 838M Nov 26 19:39 CpG_context_EF05-EM06-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 832M Nov 26 20:10 CpG_context_EF06-EM01-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 638M Nov 26 20:38 CpG_context_EF06-EM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 128M Nov 26 20:57 CpG_context_EF06-EM06-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 151M Nov 26 21:12 CpG_context_EF07-EM01-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 617M Nov 26 21:33 CpG_context_EF07-EM03-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 735M Nov 26 22:01 CpG_context_EF08-EM03-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 597M Nov 26 22:27 CpG_context_EF08-EM04-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam  75M Nov 26 16:24 EF01-EM01-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz
    -rw-r--r-- 1 sam sam 455M Nov 26 16:24 EF01-EM01-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz.bismark.zero.cov
    -rw-r--r-- 1 sam sam  70M Nov 26 16:24 EF01-EM01-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bismark.cov.gz
    -rw-r--r-- 1 sam sam 120M Nov 26 16:35 EF01-EM01-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 EF01-EM01-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  23K Dec  4 12:17 EF01-EM01-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.M-bias.txt
    -rw-r--r-- 1 sam sam  927 Dec  4 12:17 EF01-EM01-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated_splitting_report.txt
    -rw-r--r-- 1 sam sam  63M Dec  4 12:17 EF01-EM01-Zygote_R1_001.fastp-trim.CpG_report.merged_CpG_evidence.cov.gz
    -rw-r--r-- 1 sam sam 120M Nov 27 10:49 EF01-EM01-Zygote_R1_001.fastp-trim.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 EF01-EM01-Zygote_R1_001.fastp-trim.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  75M Nov 26 16:45 EF02-EM02-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz
    -rw-r--r-- 1 sam sam 463M Nov 26 16:45 EF02-EM02-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz.bismark.zero.cov
    -rw-r--r-- 1 sam sam  72M Nov 26 16:45 EF02-EM02-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bismark.cov.gz
    -rw-r--r-- 1 sam sam 120M Nov 26 16:56 EF02-EM02-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 EF02-EM02-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  22K Dec  4 12:17 EF02-EM02-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.M-bias.txt
    -rw-r--r-- 1 sam sam  925 Dec  4 12:17 EF02-EM02-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated_splitting_report.txt
    -rw-r--r-- 1 sam sam  62M Dec  4 12:17 EF02-EM02-Zygote_R1_001.fastp-trim.CpG_report.merged_CpG_evidence.cov.gz
    -rw-r--r-- 1 sam sam 120M Nov 27 11:01 EF02-EM02-Zygote_R1_001.fastp-trim.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 EF02-EM02-Zygote_R1_001.fastp-trim.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  54M Nov 26 17:02 EF03-EM03-Zygote_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.bedGraph.gz
    -rw-r--r-- 1 sam sam 327M Nov 26 17:02 EF03-EM03-Zygote_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.bedGraph.gz.bismark.zero.cov
    -rw-r--r-- 1 sam sam  50M Nov 26 17:02 EF03-EM03-Zygote_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.bismark.cov.gz
    -rw-r--r-- 1 sam sam 115M Nov 26 17:13 EF03-EM03-Zygote_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 EF03-EM03-Zygote_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  22K Dec  4 12:17 EF03-EM03-Zygote_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.M-bias.txt
    -rw-r--r-- 1 sam sam  934 Dec  4 12:17 EF03-EM03-Zygote_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated_splitting_report.txt
    -rw-r--r-- 1 sam sam  49M Dec  4 12:17 EF03-EM03-Zygote_R1_001.fastp-trim.REPAIRED.CpG_report.merged_CpG_evidence.cov.gz
    -rw-r--r-- 1 sam sam 115M Nov 27 11:12 EF03-EM03-Zygote_R1_001.fastp-trim.REPAIRED.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 EF03-EM03-Zygote_R1_001.fastp-trim.REPAIRED.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  94M Nov 26 17:28 EF03-EM04-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz
    -rw-r--r-- 1 sam sam 575M Nov 26 17:28 EF03-EM04-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz.bismark.zero.cov
    -rw-r--r-- 1 sam sam  91M Nov 26 17:28 EF03-EM04-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bismark.cov.gz
    -rw-r--r-- 1 sam sam 125M Nov 26 17:39 EF03-EM04-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 EF03-EM04-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  23K Dec  4 12:17 EF03-EM04-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.M-bias.txt
    -rw-r--r-- 1 sam sam  928 Dec  4 12:17 EF03-EM04-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated_splitting_report.txt
    -rw-r--r-- 1 sam sam  73M Dec  4 12:17 EF03-EM04-Larvae_R1_001.fastp-trim.CpG_report.merged_CpG_evidence.cov.gz
    -rw-r--r-- 1 sam sam 125M Nov 27 11:24 EF03-EM04-Larvae_R1_001.fastp-trim.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 EF03-EM04-Larvae_R1_001.fastp-trim.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  93M Nov 26 17:54 EF03-EM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz
    -rw-r--r-- 1 sam sam 572M Nov 26 17:54 EF03-EM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz.bismark.zero.cov
    -rw-r--r-- 1 sam sam  91M Nov 26 17:54 EF03-EM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bismark.cov.gz
    -rw-r--r-- 1 sam sam 125M Nov 26 18:06 EF03-EM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 EF03-EM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  23K Dec  4 12:17 EF03-EM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.M-bias.txt
    -rw-r--r-- 1 sam sam  928 Dec  4 12:17 EF03-EM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated_splitting_report.txt
    -rw-r--r-- 1 sam sam  73M Dec  4 12:17 EF03-EM05-Larvae_R1_001.fastp-trim.CpG_report.merged_CpG_evidence.cov.gz
    -rw-r--r-- 1 sam sam 125M Nov 27 11:36 EF03-EM05-Larvae_R1_001.fastp-trim.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 EF03-EM05-Larvae_R1_001.fastp-trim.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  41M Nov 26 18:09 EF04-EM04-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz
    -rw-r--r-- 1 sam sam 249M Nov 26 18:09 EF04-EM04-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz.bismark.zero.cov
    -rw-r--r-- 1 sam sam  37M Nov 26 18:09 EF04-EM04-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bismark.cov.gz
    -rw-r--r-- 1 sam sam 111M Nov 26 18:20 EF04-EM04-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 EF04-EM04-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  20K Dec  4 12:17 EF04-EM04-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.M-bias.txt
    -rw-r--r-- 1 sam sam  921 Dec  4 12:17 EF04-EM04-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated_splitting_report.txt
    -rw-r--r-- 1 sam sam  38M Dec  4 12:17 EF04-EM04-Zygote_R1_001.fastp-trim.CpG_report.merged_CpG_evidence.cov.gz
    -rw-r--r-- 1 sam sam 111M Nov 27 11:47 EF04-EM04-Zygote_R1_001.fastp-trim.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 EF04-EM04-Zygote_R1_001.fastp-trim.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  97M Nov 26 18:39 EF04-EM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz
    -rw-r--r-- 1 sam sam 597M Nov 26 18:39 EF04-EM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz.bismark.zero.cov
    -rw-r--r-- 1 sam sam  97M Nov 26 18:39 EF04-EM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bismark.cov.gz
    -rw-r--r-- 1 sam sam 127M Nov 26 18:50 EF04-EM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 EF04-EM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  23K Dec  4 12:17 EF04-EM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.M-bias.txt
    -rw-r--r-- 1 sam sam  927 Dec  4 12:17 EF04-EM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated_splitting_report.txt
    -rw-r--r-- 1 sam sam  77M Dec  4 12:17 EF04-EM05-Larvae_R1_001.fastp-trim.CpG_report.merged_CpG_evidence.cov.gz
    -rw-r--r-- 1 sam sam 127M Nov 27 11:59 EF04-EM05-Larvae_R1_001.fastp-trim.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 EF04-EM05-Larvae_R1_001.fastp-trim.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  81M Nov 26 19:02 EF05-EM01-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz
    -rw-r--r-- 1 sam sam 498M Nov 26 19:02 EF05-EM01-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz.bismark.zero.cov
    -rw-r--r-- 1 sam sam  78M Nov 26 19:02 EF05-EM01-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bismark.cov.gz
    -rw-r--r-- 1 sam sam 122M Nov 26 19:13 EF05-EM01-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 EF05-EM01-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  23K Dec  4 12:17 EF05-EM01-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.M-bias.txt
    -rw-r--r-- 1 sam sam  926 Dec  4 12:17 EF05-EM01-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated_splitting_report.txt
    -rw-r--r-- 1 sam sam  66M Dec  4 12:17 EF05-EM01-Larvae_R1_001.fastp-trim.CpG_report.merged_CpG_evidence.cov.gz
    -rw-r--r-- 1 sam sam 122M Nov 27 12:11 EF05-EM01-Larvae_R1_001.fastp-trim.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 EF05-EM01-Larvae_R1_001.fastp-trim.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  27M Nov 26 19:15 EF05-EM05-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz
    -rw-r--r-- 1 sam sam 162M Nov 26 19:15 EF05-EM05-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz.bismark.zero.cov
    -rw-r--r-- 1 sam sam  23M Nov 26 19:15 EF05-EM05-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bismark.cov.gz
    -rw-r--r-- 1 sam sam 108M Nov 26 19:25 EF05-EM05-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.7K Dec  4 12:17 EF05-EM05-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  19K Dec  4 12:17 EF05-EM05-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.M-bias.txt
    -rw-r--r-- 1 sam sam  918 Dec  4 12:17 EF05-EM05-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated_splitting_report.txt
    -rw-r--r-- 1 sam sam  26M Dec  4 12:17 EF05-EM05-Zygote_R1_001.fastp-trim.CpG_report.merged_CpG_evidence.cov.gz
    -rw-r--r-- 1 sam sam 108M Nov 27 12:22 EF05-EM05-Zygote_R1_001.fastp-trim.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.7K Dec  4 12:17 EF05-EM05-Zygote_R1_001.fastp-trim.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam 110M Nov 26 19:46 EF05-EM06-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz
    -rw-r--r-- 1 sam sam 684M Nov 26 19:46 EF05-EM06-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz.bismark.zero.cov
    -rw-r--r-- 1 sam sam 108M Nov 26 19:46 EF05-EM06-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bismark.cov.gz
    -rw-r--r-- 1 sam sam 129M Nov 26 19:58 EF05-EM06-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 EF05-EM06-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  24K Dec  4 12:17 EF05-EM06-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.M-bias.txt
    -rw-r--r-- 1 sam sam  928 Dec  4 12:17 EF05-EM06-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated_splitting_report.txt
    -rw-r--r-- 1 sam sam  83M Dec  4 12:17 EF05-EM06-Larvae_R1_001.fastp-trim.CpG_report.merged_CpG_evidence.cov.gz
    -rw-r--r-- 1 sam sam 129M Nov 27 12:34 EF05-EM06-Larvae_R1_001.fastp-trim.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 EF05-EM06-Larvae_R1_001.fastp-trim.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam 101M Nov 26 20:17 EF06-EM01-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz
    -rw-r--r-- 1 sam sam 627M Nov 26 20:17 EF06-EM01-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz.bismark.zero.cov
    -rw-r--r-- 1 sam sam 101M Nov 26 20:17 EF06-EM01-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bismark.cov.gz
    -rw-r--r-- 1 sam sam 127M Nov 26 20:28 EF06-EM01-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 EF06-EM01-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  24K Dec  4 12:17 EF06-EM01-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.M-bias.txt
    -rw-r--r-- 1 sam sam  928 Dec  4 12:17 EF06-EM01-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated_splitting_report.txt
    -rw-r--r-- 1 sam sam  79M Dec  4 12:17 EF06-EM01-Larvae_R1_001.fastp-trim.CpG_report.merged_CpG_evidence.cov.gz
    -rw-r--r-- 1 sam sam 127M Nov 27 12:46 EF06-EM01-Larvae_R1_001.fastp-trim.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 EF06-EM01-Larvae_R1_001.fastp-trim.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  88M Nov 26 20:43 EF06-EM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz
    -rw-r--r-- 1 sam sam 541M Nov 26 20:43 EF06-EM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz.bismark.zero.cov
    -rw-r--r-- 1 sam sam  86M Nov 26 20:43 EF06-EM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bismark.cov.gz
    -rw-r--r-- 1 sam sam 125M Nov 26 20:55 EF06-EM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 EF06-EM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  23K Dec  4 12:17 EF06-EM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.M-bias.txt
    -rw-r--r-- 1 sam sam  928 Dec  4 12:17 EF06-EM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated_splitting_report.txt
    -rw-r--r-- 1 sam sam  71M Dec  4 12:17 EF06-EM02-Larvae_R1_001.fastp-trim.CpG_report.merged_CpG_evidence.cov.gz
    -rw-r--r-- 1 sam sam 125M Nov 27 12:58 EF06-EM02-Larvae_R1_001.fastp-trim.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 EF06-EM02-Larvae_R1_001.fastp-trim.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  45M Nov 26 20:58 EF06-EM06-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz
    -rw-r--r-- 1 sam sam 274M Nov 26 20:58 EF06-EM06-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz.bismark.zero.cov
    -rw-r--r-- 1 sam sam  40M Nov 26 20:58 EF06-EM06-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bismark.cov.gz
    -rw-r--r-- 1 sam sam 112M Nov 26 21:09 EF06-EM06-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.7K Dec  4 12:17 EF06-EM06-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  21K Dec  4 12:17 EF06-EM06-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.M-bias.txt
    -rw-r--r-- 1 sam sam  922 Dec  4 12:17 EF06-EM06-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated_splitting_report.txt
    -rw-r--r-- 1 sam sam  42M Dec  4 12:17 EF06-EM06-Larvae_R1_001.fastp-trim.CpG_report.merged_CpG_evidence.cov.gz
    -rw-r--r-- 1 sam sam 112M Nov 27 13:09 EF06-EM06-Larvae_R1_001.fastp-trim.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.7K Dec  4 12:17 EF06-EM06-Larvae_R1_001.fastp-trim.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  48M Nov 26 21:13 EF07-EM01-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz
    -rw-r--r-- 1 sam sam 297M Nov 26 21:13 EF07-EM01-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz.bismark.zero.cov
    -rw-r--r-- 1 sam sam  44M Nov 26 21:13 EF07-EM01-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bismark.cov.gz
    -rw-r--r-- 1 sam sam 112M Nov 26 21:24 EF07-EM01-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.7K Dec  4 12:17 EF07-EM01-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  21K Dec  4 12:17 EF07-EM01-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.M-bias.txt
    -rw-r--r-- 1 sam sam  923 Dec  4 12:17 EF07-EM01-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated_splitting_report.txt
    -rw-r--r-- 1 sam sam  44M Dec  4 12:17 EF07-EM01-Zygote_R1_001.fastp-trim.CpG_report.merged_CpG_evidence.cov.gz
    -rw-r--r-- 1 sam sam 112M Nov 27 13:22 EF07-EM01-Zygote_R1_001.fastp-trim.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.7K Dec  4 12:17 EF07-EM01-Zygote_R1_001.fastp-trim.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  88M Nov 26 21:38 EF07-EM03-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz
    -rw-r--r-- 1 sam sam 547M Nov 26 21:38 EF07-EM03-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz.bismark.zero.cov
    -rw-r--r-- 1 sam sam  87M Nov 26 21:38 EF07-EM03-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bismark.cov.gz
    -rw-r--r-- 1 sam sam 124M Nov 26 21:50 EF07-EM03-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 EF07-EM03-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  23K Dec  4 12:17 EF07-EM03-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.M-bias.txt
    -rw-r--r-- 1 sam sam  927 Dec  4 12:17 EF07-EM03-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated_splitting_report.txt
    -rw-r--r-- 1 sam sam  71M Dec  4 12:17 EF07-EM03-Larvae_R1_001.fastp-trim.CpG_report.merged_CpG_evidence.cov.gz
    -rw-r--r-- 1 sam sam 124M Nov 27 13:34 EF07-EM03-Larvae_R1_001.fastp-trim.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 EF07-EM03-Larvae_R1_001.fastp-trim.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  89M Nov 26 22:06 EF08-EM03-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz
    -rw-r--r-- 1 sam sam 546M Nov 26 22:06 EF08-EM03-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz.bismark.zero.cov
    -rw-r--r-- 1 sam sam  88M Nov 26 22:06 EF08-EM03-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bismark.cov.gz
    -rw-r--r-- 1 sam sam 125M Nov 26 22:18 EF08-EM03-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 EF08-EM03-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  23K Dec  4 12:17 EF08-EM03-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.M-bias.txt
    -rw-r--r-- 1 sam sam  927 Dec  4 12:17 EF08-EM03-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated_splitting_report.txt
    -rw-r--r-- 1 sam sam  72M Dec  4 12:17 EF08-EM03-Larvae_R1_001.fastp-trim.CpG_report.merged_CpG_evidence.cov.gz
    -rw-r--r-- 1 sam sam 125M Nov 27 13:46 EF08-EM03-Larvae_R1_001.fastp-trim.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 EF08-EM03-Larvae_R1_001.fastp-trim.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  84M Nov 26 22:31 EF08-EM04-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz
    -rw-r--r-- 1 sam sam 517M Nov 26 22:31 EF08-EM04-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bedGraph.gz.bismark.zero.cov
    -rw-r--r-- 1 sam sam  82M Nov 26 22:31 EF08-EM04-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.bismark.cov.gz
    -rw-r--r-- 1 sam sam 124M Nov 26 22:43 EF08-EM04-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 EF08-EM04-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.cytosine_context_summary.txt
    -rw-r--r-- 1 sam sam  23K Dec  4 12:17 EF08-EM04-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.M-bias.txt
    -rw-r--r-- 1 sam sam  928 Dec  4 12:17 EF08-EM04-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated_splitting_report.txt
    -rw-r--r-- 1 sam sam  68M Dec  4 12:17 EF08-EM04-Larvae_R1_001.fastp-trim.CpG_report.merged_CpG_evidence.cov.gz
    -rw-r--r-- 1 sam sam 124M Nov 27 13:57 EF08-EM04-Larvae_R1_001.fastp-trim.CpG_report.txt.gz
    -rw-r--r-- 1 sam sam 1.8K Dec  4 12:17 EF08-EM04-Larvae_R1_001.fastp-trim.cytosine_context_summary.txt
    drwxr-xr-x 2 sam sam 4.0K Dec  4 12:17 multiqc_data
    -rw-r--r-- 1 sam sam 1.2M Dec  4 12:17 multiqc_report.html
    -rw-r--r-- 1 sam sam 191M Nov 26 11:16 Non_CpG_context_CF01-CM01-Zygote_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 2.6G Nov 26 11:37 Non_CpG_context_CF01-CM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 890M Nov 26 11:59 Non_CpG_context_CF02-CM02-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 1.3G Nov 26 12:18 Non_CpG_context_CF03-CM03-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 2.2G Nov 26 12:41 Non_CpG_context_CF03-CM04-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 2.4G Nov 26 13:07 Non_CpG_context_CF03-CM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 1.6G Nov 26 13:30 Non_CpG_context_CF04-CM04-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 3.4G Nov 26 13:58 Non_CpG_context_CF05-CM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 1.9G Nov 26 14:25 Non_CpG_context_CF05-CM05-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 1.6G Nov 26 14:47 Non_CpG_context_CF06-CM01-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 1.4G Nov 26 15:07 Non_CpG_context_CF06-CM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 992M Nov 26 15:25 Non_CpG_context_CF07-CM02-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 970M Nov 26 15:42 Non_CpG_context_CF08-CM03-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam  29M Nov 26 15:55 Non_CpG_context_CF08-CM04-Larvae_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam  27M Nov 26 16:05 Non_CpG_context_CF08-CM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 1.5G Nov 26 16:21 Non_CpG_context_EF01-EM01-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 1.6G Nov 26 16:42 Non_CpG_context_EF02-EM02-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 904M Nov 26 17:01 Non_CpG_context_EF03-EM03-Zygote_R1_001.fastp-trim.REPAIRED_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 2.3G Nov 26 17:23 Non_CpG_context_EF03-EM04-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 2.4G Nov 26 17:49 Non_CpG_context_EF03-EM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 485M Nov 26 18:08 Non_CpG_context_EF04-EM04-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 3.1G Nov 26 18:32 Non_CpG_context_EF04-EM05-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 1.9G Nov 26 18:58 Non_CpG_context_EF05-EM01-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 275M Nov 26 19:14 Non_CpG_context_EF05-EM05-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 3.4G Nov 26 19:39 Non_CpG_context_EF05-EM06-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 3.1G Nov 26 20:10 Non_CpG_context_EF06-EM01-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 2.3G Nov 26 20:38 Non_CpG_context_EF06-EM02-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 550M Nov 26 20:57 Non_CpG_context_EF06-EM06-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 653M Nov 26 21:12 Non_CpG_context_EF07-EM01-Zygote_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 2.4G Nov 26 21:33 Non_CpG_context_EF07-EM03-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 2.7G Nov 26 22:01 Non_CpG_context_EF08-EM03-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz
    -rw-r--r-- 1 sam sam 2.1G Nov 26 22:27 Non_CpG_context_EF08-EM04-Larvae_R1_001.fastp-trim_bismark_bt2_pe.deduplicated.txt.gz

# 7 Checksums

``` bash
# Load bash variables into memory
source .bashvars

cd "${output_dir_top}"

for file in *; do
  if [ "${file}" != "checksums.md5" ]; then
    md5sum "${file}" >> checksums.md5
  fi
done
```

# 8 MultiQC

``` bash
# Load bash variables into memory
source .bashvars

cd "${output_dir_top}"

${multiqc} .
```

<div id="refs" class="references csl-bib-body hanging-indent">

<div id="ref-ewels2016" class="csl-entry">

Ewels, Philip, Måns Magnusson, Sverker Lundin, and Max Käller. 2016.
“MultiQC: Summarize Analysis Results for Multiple Tools and Samples in a
Single Report.” *Bioinformatics* 32 (19): 3047–48.
<https://doi.org/10.1093/bioinformatics/btw354>.

</div>

<div id="ref-krueger2011" class="csl-entry">

Krueger, Felix, and Simon R. Andrews. 2011. “Bismark: A Flexible Aligner
and Methylation Caller for Bisulfite-Seq Applications.” *Bioinformatics*
27 (11): 1571–72. <https://doi.org/10.1093/bioinformatics/btr167>.

</div>

</div>
